{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# start docker deamon in terminal\n",
    "# !sudo docker login\n",
    "# OR - becuase the jupyther with allow root\n",
    "# !docker login"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NA_rna_qc_plot.py\t  NA_scrna_qc_metrics-Copy1.py\tscrna_plot_qc_metrics.R\n",
      "NA_scrna_plot_qc_debug.R  barcode_rank_functions.R\tscrna_qc_metrics.py\n"
     ]
    }
   ],
   "source": [
    "!ls bin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !chmod +x ./bin/scrna_qc_metrics.py\n",
    "# !chmod +x /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/bin/scrna_plot_qc_metrics.R\n",
    "\n",
    "# !chmod +x /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/bin/scrna_plot_qc_debug.R\n",
    "# !chmod +x /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/bin/scrna_plot_qc_metrics.R\n",
    "\n",
    "# !chmod +x /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/bin/barcode_rank_functions.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ls: cannot access '/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/bin/scrna_plot_qc_debug.R': No such file or directory\n"
     ]
    }
   ],
   "source": [
    "!ls /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/bin/scrna_plot_qc_debug.R\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  Credentialed Accounts\n",
      "ACTIVE  ACCOUNT\n",
      "        886987709752-compute@developer.gserviceaccount.com\n",
      "*       eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\n",
      "\n",
      "To set the active account, run:\n",
      "    $ gcloud config set account `ACCOUNT`\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!gcloud auth list\n",
    "# gcloud config set account\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "# os.environ[\"TOWER_ACCESS_TOKEN\"] = 'XXX'\n",
    "# os.environ[\"TOWER_ACCESS_TOKEN\"] = 'eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "os.environ[\"TOWER_ACCESS_TOKEN\"] = 'eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "os.environ[\"TOWER_WORKSPACE_ID\"]='247507587970818'\n",
    "\n",
    "!echo $TOWER_ACCESS_TOKEN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Version: 23.10.0 build 5889\n",
      "  Created: 15-10-2023 15:07 UTC \n",
      "  System: Linux 5.10.0-26-cloud-amd64\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 11.0.13+7-b1751.21\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna'"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Overwriting nextflow_local.config\n"
     ]
    }
   ],
   "source": [
    "%%writefile nextflow_local.config\n",
    "\n",
    "includeConfig 'conf/conda.config'\n",
    "\n",
    "\n",
    "manifest {\n",
    "  description = 'A nexflow pipeline for fg single cell' \n",
    "  mainScript = 'nf_wf_fg_singlecell_rna.nf'\n",
    "}\n",
    "\n",
    "profiles {\n",
    "  conda_profile {\n",
    "    includeConfig 'conf/conda.config'\n",
    "    conda.enabled = true\n",
    "    executor = 'local'\n",
    "    }\n",
    "\n",
    "  conda_docker_profile {\n",
    "    includeConfig 'conf/conda_docker.config'\n",
    "    executor = 'local'\n",
    "    }\n",
    "\n",
    "  docker_local {\n",
    "    includeConfig 'conf/docker.config'\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "  }\n",
    "  google_batch_profile {\n",
    "    workDir = 'gs://eila-nextflow-bucket/work/RNA'\n",
    "    wave.enabled = true\n",
    "    fusion.enabled = true\n",
    "    process.scratch = false\n",
    "    includeConfig 'conf/docker-google-batch.config'\n",
    "    docker.enabled = true\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    google.project = 'hai-gcp-genomic'\n",
    "    google.location = 'us-central1'\n",
    "    google.batch.spot = false\n",
    "    google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "  }\n",
    "}\n",
    "\n",
    "// Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "params.ORGANISM = 'human'\n",
    "params.FASTQS_SPEC_CH = '/Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/rna_fastqs_files_local.csv'\n",
    "params.GENOME_FASTA='http://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz'\n",
    "params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "params.ENV_SYNAPSE_TOKEN = \"nextflow secrets get SYNAPSE\".execute().text.trim()\n",
    "params.SYNAPSE_CONFIG = '/Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/synapse.config'\n",
    "params.SYNAPSE_IDS = '/Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/synapse_ids.csv'\n",
    "params.SC_RNA_QC_SCRIPT='scrna_qc_metrics.py'\n",
    "params.SC_RNA_QC_PLOT_SCRIPT='scrna_plot_qc_debug.R'\n",
    "params.SC_RNA_QC_PLOT_HELPER_SCRIPT='barcode_rank_functions.R'\n",
    "params.SUBPOOL = 'no_subpool'\n",
    "params.UMI_CUTOFF=2\n",
    "params.GENE_CUTOFF=3\n",
    "params.RNA_DEBUG_QC='/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/sc_rna_qc_calculate.tsv'\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# params.ANNDATA_DEBUG_FILE = '/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/adata.h5ad'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rw-r--r-- 1 eila eila 391817928 Nov 13 23:23 /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/adata.h5ad\n"
     ]
    }
   ],
   "source": [
    "!ls -lt /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/adata.h5ad"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Version: 23.10.0 build 5889\n",
      "  Created: 15-10-2023 15:07 UTC \n",
      "  System: Linux 5.10.0-26-cloud-amd64\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 11.0.13+7-b1751.21\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !ls /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Overwriting nextflow_vm.config\n"
     ]
    }
   ],
   "source": [
    "%%writefile nextflow_vm.config\n",
    "\n",
    "includeConfig 'conf/conda.config'\n",
    "\n",
    "\n",
    "manifest {\n",
    "  description = 'A nexflow pipeline for fg single cell' \n",
    "  mainScript = 'nf_wf_fg_singlecell_rna.nf'\n",
    "}\n",
    "\n",
    "profiles {\n",
    "  conda_profile {\n",
    "    includeConfig 'conf/conda.config'\n",
    "    conda.enabled = true\n",
    "    executor = 'local'\n",
    "    }\n",
    "\n",
    "  docker_local {\n",
    "    includeConfig 'conf/docker.config'\n",
    "    docker.enabled = true\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    executor = 'local'\n",
    "  }\n",
    "  google_batch_profile {\n",
    "    workDir = 'gs://eila-nextflow-bucket/work/RNA'\n",
    "    includeConfig 'conf/docker-google-batch.config'\n",
    "    docker.enabled = true\n",
    "    wave.enabled = true\n",
    "    fusion.enabled = true\n",
    "    process.scratch = false\n",
    "    executor = 'google-batch'\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    google.project = 'hai-gcp-genomic'\n",
    "    google.location = 'us-central1'\n",
    "    google.batch.spot = true\n",
    "    google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "  }\n",
    "    \n",
    "}\n",
    "\n",
    "// Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "params.ORGANISM = 'human'\n",
    "params.FASTQS_SPEC_CH = '/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/rna_fastqs_files_vm-dogma-dig.csv'\n",
    "params.GENOME_FASTA='http://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz'\n",
    "params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "params.ENV_SYNAPSE_TOKEN = \"nextflow secrets get SYNAPSE\".execute().text.trim()\n",
    "params.SYNAPSE_CONFIG = '/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/synapse.config'\n",
    "params.SYNAPSE_IDS = '/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/synapse_ids.csv'\n",
    "params.SUBPOOL = 'no_subpool'\n",
    "params.UMI_CUTOFF=2\n",
    "params.GENE_CUTOFF=3\n",
    "params.SC_RNA_QC_SCRIPT='scrna_qc_metrics.py'\n",
    "params.SC_RNA_QC_PLOT_SCRIPT='scrna_plot_qc_metrics.R'\n",
    "params.SC_RNA_QC_PLOT_HELPER_SCRIPT='barcode_rank_functions.R'\n",
    "params.UMI_CUTOFF=2\n",
    "params.GENE_CUTOFF=3\n",
    "params.RNA_DEBUG_QC='/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/sc_rna_qc_calculate.tsv'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# params.ANNDATA_FILE = '/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/adata.h5ad'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !cat /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/rna_fastqs_files_vm-dogma-dig.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !head /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/nf_spec_dogmaseq_dig.yaml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !head /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/RNA-737K-arc-v1.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !ls /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm -rf /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/work\n",
    "\n",
    "\n",
    "# !gsutil -m rm -r gs://eila-nextflow-bucket/work/RNA\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.10.0\n",
      "Launching `nf_wf_fg_singlecell_rna.nf` [jolly_hopper] DSL2 - revision: fe1c2abe14\n",
      "Nov 17, 2023 1:47:11 AM com.google.auth.oauth2.DefaultCredentialsProvider warnAboutProblematicCredentials\n",
      "WARNING: Your application has authenticated using end user credentials from Google Cloud SDK. We recommend that most server applications use service accounts instead. If your application continues to use end user credentials from Cloud SDK, you might receive a \"quota exceeded\" or \"API not enabled\" error. For more information about service accounts, see https://cloud.google.com/docs/authentication/.\n",
      "null\n",
      "/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/rna_fastqs_files_vm-dogma-dig.csv\n",
      "[-        ] process > run_seqspec_modify_rna     -\u001b[K\n",
      "[-        ] process > run_seqspec_print          -\u001b[K\n",
      "[-        ] process > run_seqspec_index_rna_kb   -\u001b[K\n",
      "[-        ] process > run_kb_ref                 -\u001b[K\n",
      "[-        ] process > run_kb_count               -\u001b[K\n",
      "[-        ] process > scrna_calculate_qc_metrics -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_seqspec_index_rna_kb\u001b[K\n",
      "technology is $technology\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after run_kb_ref\u001b[K\n",
      "after get organism\u001b[K\n",
      "after run_kb_count\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "sc_rna_qc_plot_script is: scrna_plot_qc_metrics.R\u001b[K\n",
      "sc_rna_qc_plot_helper_script is: barcode_rank_functions.R\u001b[K\n",
      "\u001b[20A\n",
      "[-        ] process > run_seqspec_modify_rna     -\u001b[K\n",
      "[-        ] process > run_seqspec_print          -\u001b[K\n",
      "[-        ] process > run_seqspec_index_rna_kb   -\u001b[K\n",
      "[-        ] process > run_kb_ref                 -\u001b[K\n",
      "[-        ] process > run_kb_count               -\u001b[K\n",
      "[-        ] process > scrna_calculate_qc_metrics -\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_seqspec_index_rna_kb\u001b[K\n",
      "technology is $technology\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after run_kb_ref\u001b[K\n",
      "after get organism\u001b[K\n",
      "after run_kb_count\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "sc_rna_qc_plot_script is: scrna_plot_qc_metrics.R\u001b[K\n",
      "sc_rna_qc_plot_helper_script is: barcode_rank_functions.R\u001b[K\n",
      "\u001b[21A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_seqspec_modify_rna     -\u001b[K\n",
      "[-        ] process > run_seqspec_print          -\u001b[K\n",
      "[-        ] process > run_seqspec_index_rna_kb   -\u001b[K\n",
      "[-        ] process > run_kb_ref                 -\u001b[K\n",
      "[-        ] process > run_kb_count               -\u001b[K\n",
      "[-        ] process > scrna_calculate_qc_metrics -\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_seqspec_index_rna_kb\u001b[K\n",
      "technology is $technology\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after run_kb_ref\u001b[K\n",
      "after get organism\u001b[K\n",
      "after run_kb_count\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "sc_rna_qc_plot_script is: scrna_plot_qc_metrics.R\u001b[K\n",
      "sc_rna_qc_plot_helper_script is: barcode_rank_functions.R\u001b[K\n",
      "\u001b[22A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1) [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_seqspec_print          -\u001b[K\n",
      "[-        ] process > run_seqspec_index_rna_kb   -\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                 [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_kb_count               -\u001b[K\n",
      "[-        ] process > scrna_calculate_qc_metrics -\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_seqspec_index_rna_kb\u001b[K\n",
      "technology is $technology\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after run_kb_ref\u001b[K\n",
      "after get organism\u001b[K\n",
      "after run_kb_count\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "sc_rna_qc_plot_script is: scrna_plot_qc_metrics.R\u001b[K\n",
      "sc_rna_qc_plot_helper_script is: barcode_rank_functions.R\u001b[K\n",
      "\u001b[22A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1) [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_seqspec_print          -\u001b[K\n",
      "[-        ] process > run_seqspec_index_rna_kb   -\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                 [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_kb_count               -\u001b[K\n",
      "[-        ] process > scrna_calculate_qc_metrics -\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_seqspec_index_rna_kb\u001b[K\n",
      "technology is $technology\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after run_kb_ref\u001b[K\n",
      "after get organism\u001b[K\n",
      "after run_kb_count\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "sc_rna_qc_plot_script is: scrna_plot_qc_metrics.R\u001b[K\n",
      "sc_rna_qc_plot_helper_script is: barcode_rank_functions.R\u001b[K\n",
      "dna_assempbly_fa_gz is Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "gtf_gz is Homo_sapiens.GRCh38.109.gtf.gz\u001b[K\n",
      "finished callling kb ref\u001b[K\n",
      "\u001b[K\n",
      "\u001b[26A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_seqspec_print          -\u001b[K\n",
      "[-        ] process > run_seqspec_index_rna_kb   -\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                 [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_kb_count               -\u001b[K\n",
      "[-        ] process > scrna_calculate_qc_metrics -\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_seqspec_index_rna_kb\u001b[K\n",
      "technology is $technology\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after run_kb_ref\u001b[K\n",
      "after get organism\u001b[K\n",
      "after run_kb_count\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "sc_rna_qc_plot_script is: scrna_plot_qc_metrics.R\u001b[K\n",
      "sc_rna_qc_plot_helper_script is: barcode_rank_functions.R\u001b[K\n",
      "dna_assempbly_fa_gz is Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "gtf_gz is Homo_sapiens.GRCh38.109.gtf.gz\u001b[K\n",
      "finished callling kb ref\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_rna\u001b[K\n",
      "SRR18677638_1.fastq.gz\u001b[K\n",
      "SRR18677638_2.fastq.gz\u001b[K\n",
      "na.na\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify rna\u001b[K\n",
      "\u001b[K\n",
      "\u001b[33A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[77/6b6ef6] process > run_seqspec_print (1)      [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_seqspec_index_rna_kb   -\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                 [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_kb_count               -\u001b[K\n",
      "[-        ] process > scrna_calculate_qc_metrics -\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_seqspec_index_rna_kb\u001b[K\n",
      "technology is $technology\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after run_kb_ref\u001b[K\n",
      "after get organism\u001b[K\n",
      "after run_kb_count\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "sc_rna_qc_plot_script is: scrna_plot_qc_metrics.R\u001b[K\n",
      "sc_rna_qc_plot_helper_script is: barcode_rank_functions.R\u001b[K\n",
      "dna_assempbly_fa_gz is Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "gtf_gz is Homo_sapiens.GRCh38.109.gtf.gz\u001b[K\n",
      "finished callling kb ref\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_rna\u001b[K\n",
      "SRR18677638_1.fastq.gz\u001b[K\n",
      "SRR18677638_2.fastq.gz\u001b[K\n",
      "na.na\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify rna\u001b[K\n",
      "\u001b[K\n",
      "\u001b[33A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[77/6b6ef6] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b2/06d00c] process > run_seqspec_index_rna_kb (1) [100%] 1 of 1, cached: 1\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_kb_count                 -\u001b[K\n",
      "[-        ] process > scrna_calculate_qc_metrics   -\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs   -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_seqspec_index_rna_kb\u001b[K\n",
      "technology is $technology\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after run_kb_ref\u001b[K\n",
      "after get organism\u001b[K\n",
      "after run_kb_count\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "sc_rna_qc_plot_script is: scrna_plot_qc_metrics.R\u001b[K\n",
      "sc_rna_qc_plot_helper_script is: barcode_rank_functions.R\u001b[K\n",
      "dna_assempbly_fa_gz is Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "gtf_gz is Homo_sapiens.GRCh38.109.gtf.gz\u001b[K\n",
      "finished callling kb ref\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_rna\u001b[K\n",
      "SRR18677638_1.fastq.gz\u001b[K\n",
      "SRR18677638_2.fastq.gz\u001b[K\n",
      "na.na\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify rna\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: af70aa0b1289\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[77/6b6ef6] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b2/06d00c] process > run_seqspec_index_rna_kb (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_kb_count                 -\u001b[K\n",
      "[-        ] process > scrna_calculate_qc_metrics   -\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs   -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_seqspec_index_rna_kb\u001b[K\n",
      "technology is $technology\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after run_kb_ref\u001b[K\n",
      "after get organism\u001b[K\n",
      "after run_kb_count\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "sc_rna_qc_plot_script is: scrna_plot_qc_metrics.R\u001b[K\n",
      "sc_rna_qc_plot_helper_script is: barcode_rank_functions.R\u001b[K\n",
      "dna_assempbly_fa_gz is Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "gtf_gz is Homo_sapiens.GRCh38.109.gtf.gz\u001b[K\n",
      "finished callling kb ref\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_rna\u001b[K\n",
      "SRR18677638_1.fastq.gz\u001b[K\n",
      "SRR18677638_2.fastq.gz\u001b[K\n",
      "na.na\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify rna\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: af70aa0b1289\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "spec_yaml is nf_seqspec_modify.yaml\u001b[K\n",
      "RNA_R1_fastq_gz is SRR18677638_1.fastq.gz\u001b[K\n",
      "RNA_R2_fastq_gz is SRR18677638_2.fastq.gz\u001b[K\n",
      "RNA_R3_fastq_gz is na.na\u001b[K\n",
      "0,0,16:0,16,28:1,0,102\u001b[K\n",
      "finished run_seqspec_index_rna\u001b[K\n",
      "\u001b[K\n",
      "\u001b[48A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[77/6b6ef6] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b2/06d00c] process > run_seqspec_index_rna_kb (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/090813] process > run_kb_count (1)             [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > scrna_calculate_qc_metrics   -\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs   -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_seqspec_index_rna_kb\u001b[K\n",
      "technology is $technology\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after run_kb_ref\u001b[K\n",
      "after get organism\u001b[K\n",
      "after run_kb_count\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "sc_rna_qc_plot_script is: scrna_plot_qc_metrics.R\u001b[K\n",
      "sc_rna_qc_plot_helper_script is: barcode_rank_functions.R\u001b[K\n",
      "dna_assempbly_fa_gz is Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "gtf_gz is Homo_sapiens.GRCh38.109.gtf.gz\u001b[K\n",
      "finished callling kb ref\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_rna\u001b[K\n",
      "SRR18677638_1.fastq.gz\u001b[K\n",
      "SRR18677638_2.fastq.gz\u001b[K\n",
      "na.na\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify rna\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: af70aa0b1289\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "spec_yaml is nf_seqspec_modify.yaml\u001b[K\n",
      "RNA_R1_fastq_gz is SRR18677638_1.fastq.gz\u001b[K\n",
      "RNA_R2_fastq_gz is SRR18677638_2.fastq.gz\u001b[K\n",
      "RNA_R3_fastq_gz is na.na\u001b[K\n",
      "0,0,16:0,16,28:1,0,102\u001b[K\n",
      "finished run_seqspec_index_rna\u001b[K\n",
      "\u001b[K\n",
      "\u001b[48A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[77/6b6ef6] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b2/06d00c] process > run_seqspec_index_rna_kb (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/090813] process > run_kb_count (1)             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > scrna_calculate_qc_metrics   -\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs   -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_seqspec_index_rna_kb\u001b[K\n",
      "technology is $technology\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after run_kb_ref\u001b[K\n",
      "after get organism\u001b[K\n",
      "after run_kb_count\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "sc_rna_qc_plot_script is: scrna_plot_qc_metrics.R\u001b[K\n",
      "sc_rna_qc_plot_helper_script is: barcode_rank_functions.R\u001b[K\n",
      "dna_assempbly_fa_gz is Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "gtf_gz is Homo_sapiens.GRCh38.109.gtf.gz\u001b[K\n",
      "finished callling kb ref\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_rna\u001b[K\n",
      "SRR18677638_1.fastq.gz\u001b[K\n",
      "SRR18677638_2.fastq.gz\u001b[K\n",
      "na.na\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify rna\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: af70aa0b1289\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "spec_yaml is nf_seqspec_modify.yaml\u001b[K\n",
      "RNA_R1_fastq_gz is SRR18677638_1.fastq.gz\u001b[K\n",
      "RNA_R2_fastq_gz is SRR18677638_2.fastq.gz\u001b[K\n",
      "RNA_R3_fastq_gz is na.na\u001b[K\n",
      "0,0,16:0,16,28:1,0,102\u001b[K\n",
      "finished run_seqspec_index_rna\u001b[K\n",
      "\u001b[K\n",
      "start run_kb_count !!!\u001b[K\n",
      "index.idx\u001b[K\n",
      "t2g.txt\u001b[K\n",
      "SRR18677638_1.fastq.gz\u001b[K\n",
      "SRR18677638_2.fastq.gz\u001b[K\n",
      "seqspec.technology.out\u001b[K\n",
      "0,0,16:0,16,28:1,0,102\u001b[K\n",
      "finished run_kb_count\u001b[K\n",
      "\u001b[K\n",
      "\u001b[57A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[77/6b6ef6] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b2/06d00c] process > run_seqspec_index_rna_kb (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/090813] process > run_kb_count (1)             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > scrna_calculate_qc_metrics   [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs   -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_seqspec_index_rna_kb\u001b[K\n",
      "technology is $technology\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after run_kb_ref\u001b[K\n",
      "after get organism\u001b[K\n",
      "after run_kb_count\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "sc_rna_qc_plot_script is: scrna_plot_qc_metrics.R\u001b[K\n",
      "sc_rna_qc_plot_helper_script is: barcode_rank_functions.R\u001b[K\n",
      "dna_assempbly_fa_gz is Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "gtf_gz is Homo_sapiens.GRCh38.109.gtf.gz\u001b[K\n",
      "finished callling kb ref\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_rna\u001b[K\n",
      "SRR18677638_1.fastq.gz\u001b[K\n",
      "SRR18677638_2.fastq.gz\u001b[K\n",
      "na.na\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify rna\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: af70aa0b1289\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "spec_yaml is nf_seqspec_modify.yaml\u001b[K\n",
      "RNA_R1_fastq_gz is SRR18677638_1.fastq.gz\u001b[K\n",
      "RNA_R2_fastq_gz is SRR18677638_2.fastq.gz\u001b[K\n",
      "RNA_R3_fastq_gz is na.na\u001b[K\n",
      "0,0,16:0,16,28:1,0,102\u001b[K\n",
      "finished run_seqspec_index_rna\u001b[K\n",
      "\u001b[K\n",
      "start run_kb_count !!!\u001b[K\n",
      "index.idx\u001b[K\n",
      "t2g.txt\u001b[K\n",
      "SRR18677638_1.fastq.gz\u001b[K\n",
      "SRR18677638_2.fastq.gz\u001b[K\n",
      "seqspec.technology.out\u001b[K\n",
      "0,0,16:0,16,28:1,0,102\u001b[K\n",
      "finished run_kb_count\u001b[K\n",
      "\u001b[K\n",
      "\u001b[57A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[77/6b6ef6] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b2/06d00c] process > run_seqspec_index_rna_kb (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/090813] process > run_kb_count (1)               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/8fd5ce] process > scrna_calculate_qc_metrics (1) [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs     -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_seqspec_index_rna_kb\u001b[K\n",
      "technology is $technology\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after run_kb_ref\u001b[K\n",
      "after get organism\u001b[K\n",
      "after run_kb_count\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "sc_rna_qc_plot_script is: scrna_plot_qc_metrics.R\u001b[K\n",
      "sc_rna_qc_plot_helper_script is: barcode_rank_functions.R\u001b[K\n",
      "dna_assempbly_fa_gz is Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "gtf_gz is Homo_sapiens.GRCh38.109.gtf.gz\u001b[K\n",
      "finished callling kb ref\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_rna\u001b[K\n",
      "SRR18677638_1.fastq.gz\u001b[K\n",
      "SRR18677638_2.fastq.gz\u001b[K\n",
      "na.na\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify rna\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: af70aa0b1289\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "spec_yaml is nf_seqspec_modify.yaml\u001b[K\n",
      "RNA_R1_fastq_gz is SRR18677638_1.fastq.gz\u001b[K\n",
      "RNA_R2_fastq_gz is SRR18677638_2.fastq.gz\u001b[K\n",
      "RNA_R3_fastq_gz is na.na\u001b[K\n",
      "0,0,16:0,16,28:1,0,102\u001b[K\n",
      "finished run_seqspec_index_rna\u001b[K\n",
      "\u001b[K\n",
      "start run_kb_count !!!\u001b[K\n",
      "index.idx\u001b[K\n",
      "t2g.txt\u001b[K\n",
      "SRR18677638_1.fastq.gz\u001b[K\n",
      "SRR18677638_2.fastq.gz\u001b[K\n",
      "seqspec.technology.out\u001b[K\n",
      "0,0,16:0,16,28:1,0,102\u001b[K\n",
      "finished run_kb_count\u001b[K\n",
      "\u001b[K\n",
      "\u001b[58A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[77/6b6ef6] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b2/06d00c] process > run_seqspec_index_rna_kb (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/090813] process > run_kb_count (1)               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/8fd5ce] process > scrna_calculate_qc_metrics (1) [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs     -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[58A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[77/6b6ef6] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b2/06d00c] process > run_seqspec_index_rna_kb (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/090813] process > run_kb_count (1)               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/8fd5ce] process > scrna_calculate_qc_metrics (1) [  0%] 0 of 1 ✔\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs     -\u001b[K\n",
      "Conda environment: # conda environments:\u001b[K\n",
      "#\u001b[K\n",
      "base                     /opt/conda\u001b[K\n",
      "rna_qc_env               /opt/conda/envs/rna_qc_env\u001b[K\n",
      "Script path: /usr/local/bin/scrna_qc_metrics.py\u001b[K\n",
      "Adata path: adata.h5ad\u001b[K\n",
      "Subpool argument: --subpool no_subpool\u001b[K\n",
      "Starting scrna_calculate_qc_metrics...\u001b[K\n",
      "Result file path: \u001b[K\n",
      "Finished scrna_calculate_qc_metrics.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[58A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[77/6b6ef6] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b2/06d00c] process > run_seqspec_index_rna_kb (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/090813] process > run_kb_count (1)               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/8fd5ce] process > scrna_calculate_qc_metrics (1) [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs     -\u001b[K\n",
      "Conda environment: # conda environments:\u001b[K\n",
      "#\u001b[K\n",
      "base                     /opt/conda\u001b[K\n",
      "rna_qc_env               /opt/conda/envs/rna_qc_env\u001b[K\n",
      "Script path: /usr/local/bin/scrna_qc_metrics.py\u001b[K\n",
      "Adata path: adata.h5ad\u001b[K\n",
      "Subpool argument: --subpool no_subpool\u001b[K\n",
      "Starting scrna_calculate_qc_metrics...\u001b[K\n",
      "Result file path: \u001b[K\n",
      "Finished scrna_calculate_qc_metrics.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[21A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[77/6b6ef6] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b2/06d00c] process > run_seqspec_index_rna_kb (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/090813] process > run_kb_count (1)               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/8fd5ce] process > scrna_calculate_qc_metrics (1) [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > scrna_plot_qc_metrics_prcs     [  0%] 0 of 1\u001b[K\n",
      "Conda environment: # conda environments:\u001b[K\n",
      "#\u001b[K\n",
      "base                     /opt/conda\u001b[K\n",
      "rna_qc_env               /opt/conda/envs/rna_qc_env\u001b[K\n",
      "Script path: /usr/local/bin/scrna_qc_metrics.py\u001b[K\n",
      "Adata path: adata.h5ad\u001b[K\n",
      "Subpool argument: --subpool no_subpool\u001b[K\n",
      "Starting scrna_calculate_qc_metrics...\u001b[K\n",
      "Result file path: \u001b[K\n",
      "Finished scrna_calculate_qc_metrics.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[21A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[77/6b6ef6] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b2/06d00c] process > run_seqspec_index_rna_kb (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/090813] process > run_kb_count (1)               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/8fd5ce] process > scrna_calculate_qc_metrics (1) [100%] 1 of 1 ✔\u001b[K\n",
      "[2a/1154df] process > scrna_plot_qc_metrics_prcs (1) [  0%] 0 of 1\u001b[K\n",
      "Conda environment: # conda environments:\u001b[K\n",
      "#\u001b[K\n",
      "base                     /opt/conda\u001b[K\n",
      "rna_qc_env               /opt/conda/envs/rna_qc_env\u001b[K\n",
      "Script path: /usr/local/bin/scrna_qc_metrics.py\u001b[K\n",
      "Adata path: adata.h5ad\u001b[K\n",
      "Subpool argument: --subpool no_subpool\u001b[K\n",
      "Starting scrna_calculate_qc_metrics...\u001b[K\n",
      "Result file path: \u001b[K\n",
      "Finished scrna_calculate_qc_metrics.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[21A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[77/6b6ef6] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b2/06d00c] process > run_seqspec_index_rna_kb (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/090813] process > run_kb_count (1)               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/8fd5ce] process > scrna_calculate_qc_metrics (1) [100%] 1 of 1 ✔\u001b[K\n",
      "[2a/1154df] process > scrna_plot_qc_metrics_prcs (1) [  0%] 0 of 1\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[21A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[77/6b6ef6] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b2/06d00c] process > run_seqspec_index_rna_kb (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/090813] process > run_kb_count (1)               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/8fd5ce] process > scrna_calculate_qc_metrics (1) [100%] 1 of 1 ✔\u001b[K\n",
      "[2a/1154df] process > scrna_plot_qc_metrics_prcs (1) [  0%] 0 of 1 ✔\u001b[K\n",
      "R QC Plot Script: scrna_plot_qc_metrics.R\u001b[K\n",
      "R QC Plot Helper Script: barcode_rank_functions.R\u001b[K\n",
      "RNA QC Metrics TSV: sc_rna_qc_calculate.tsv\u001b[K\n",
      "UMI Cutoff: 2\u001b[K\n",
      "Gene Cutoff: 3\u001b[K\n",
      "UMI Rank Plot All Output: umi_rank_plot_all_output.png\u001b[K\n",
      "UMI Rank Plot Top Output: umi_rank_plot_top_output.png\u001b[K\n",
      "Gene UMI Plot File Output: gene_umi_plot_file_output.png\u001b[K\n",
      "Starting scrna_plot_qc_metrics_prcs...\u001b[K\n",
      "Script path: /usr/local/bin/scrna_plot_qc_metrics.R\u001b[K\n",
      "Executing main script...\u001b[K\n",
      "Found file: /usr/local/bin/barcode_rank_functions.R \u001b[K\n",
      "Hello, this is the main function!\u001b[K\n",
      "Barcode Metadata File: sc_rna_qc_calculate.tsv \u001b[K\n",
      "UMI Cutoff: 2 \u001b[K\n",
      "Gene Cutoff: 3 \u001b[K\n",
      "UMI Rank Plot File: umi_rank_plot_all_output.png \u001b[K\n",
      "GENE Rank Plot File: umi_rank_plot_top_output.png \u001b[K\n",
      "GENE UMI Rank Plot File: gene_umi_plot_file_output.png \u001b[K\n",
      "current_directory: /fusion/gs/eila-nextflow-bucket/work/RNA/2a/1154df225ae96d7d4b8c2c6c6ddad0 \u001b[K\n",
      "Files in the working directory:\u001b[K\n",
      "sc_rna_qc_calculate.tsv\u001b[K\n",
      "PATH: /opt/conda/envs/rna_qc_env/bin:/opt/conda/condabin:/opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \u001b[K\n",
      "Printing RNA QC metrics...\u001b[K\n",
      "print_rna_qc_metrics: Parsed Arguments:\u001b[K\n",
      "print_rna_qc_metrics: Barcode Metadata File: sc_rna_qc_calculate.tsv \u001b[K\n",
      "print_rna_qc_metrics: UMI Cutoff: 2 \u001b[K\n",
      "print_rna_qc_metrics: Gene Cutoff: 3 \u001b[K\n",
      "print_rna_qc_metrics: UMI Rank Plot File: umi_rank_plot_all_output.png \u001b[K\n",
      "print_rna_qc_metrics: GENE Rank Plot File: umi_rank_plot_top_output.png \u001b[K\n",
      "print_rna_qc_metrics: GENE UMI Rank Plot File: gene_umi_plot_file_output.png \u001b[K\n",
      "print_rna_qc_metrics: Reading barcode metadata from file: sc_rna_qc_calculate.tsv \u001b[K\n",
      "print_rna_qc_metrics: calling get_elbow_knee_points \u001b[K\n",
      "[1] \"Returning sliced vectors\"\u001b[K\n",
      "[1] \"Returning sliced vectors\"\u001b[K\n",
      "null device \u001b[K\n",
      "          1 \u001b[K\n",
      "\u001b[K\n",
      "Finished scrna_plot_qc_metrics_prcs.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[50A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[39/e50e47] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[77/6b6ef6] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b2/06d00c] process > run_seqspec_index_rna_kb (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[04/b164fb] process > run_kb_ref                     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/090813] process > run_kb_count (1)               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[41/8fd5ce] process > scrna_calculate_qc_metrics (1) [100%] 1 of 1 ✔\u001b[K\n",
      "[2a/1154df] process > scrna_plot_qc_metrics_prcs (1) [100%] 1 of 1 ✔\u001b[K\n",
      "R QC Plot Script: scrna_plot_qc_metrics.R\u001b[K\n",
      "R QC Plot Helper Script: barcode_rank_functions.R\u001b[K\n",
      "RNA QC Metrics TSV: sc_rna_qc_calculate.tsv\u001b[K\n",
      "UMI Cutoff: 2\u001b[K\n",
      "Gene Cutoff: 3\u001b[K\n",
      "UMI Rank Plot All Output: umi_rank_plot_all_output.png\u001b[K\n",
      "UMI Rank Plot Top Output: umi_rank_plot_top_output.png\u001b[K\n",
      "Gene UMI Plot File Output: gene_umi_plot_file_output.png\u001b[K\n",
      "Starting scrna_plot_qc_metrics_prcs...\u001b[K\n",
      "Script path: /usr/local/bin/scrna_plot_qc_metrics.R\u001b[K\n",
      "Executing main script...\u001b[K\n",
      "Found file: /usr/local/bin/barcode_rank_functions.R \u001b[K\n",
      "Hello, this is the main function!\u001b[K\n",
      "Barcode Metadata File: sc_rna_qc_calculate.tsv \u001b[K\n",
      "UMI Cutoff: 2 \u001b[K\n",
      "Gene Cutoff: 3 \u001b[K\n",
      "UMI Rank Plot File: umi_rank_plot_all_output.png \u001b[K\n",
      "GENE Rank Plot File: umi_rank_plot_top_output.png \u001b[K\n",
      "GENE UMI Rank Plot File: gene_umi_plot_file_output.png \u001b[K\n",
      "current_directory: /fusion/gs/eila-nextflow-bucket/work/RNA/2a/1154df225ae96d7d4b8c2c6c6ddad0 \u001b[K\n",
      "Files in the working directory:\u001b[K\n",
      "sc_rna_qc_calculate.tsv\u001b[K\n",
      "PATH: /opt/conda/envs/rna_qc_env/bin:/opt/conda/condabin:/opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \u001b[K\n",
      "Printing RNA QC metrics...\u001b[K\n",
      "print_rna_qc_metrics: Parsed Arguments:\u001b[K\n",
      "print_rna_qc_metrics: Barcode Metadata File: sc_rna_qc_calculate.tsv \u001b[K\n",
      "print_rna_qc_metrics: UMI Cutoff: 2 \u001b[K\n",
      "print_rna_qc_metrics: Gene Cutoff: 3 \u001b[K\n",
      "print_rna_qc_metrics: UMI Rank Plot File: umi_rank_plot_all_output.png \u001b[K\n",
      "print_rna_qc_metrics: GENE Rank Plot File: umi_rank_plot_top_output.png \u001b[K\n",
      "print_rna_qc_metrics: GENE UMI Rank Plot File: gene_umi_plot_file_output.png \u001b[K\n",
      "print_rna_qc_metrics: Reading barcode metadata from file: sc_rna_qc_calculate.tsv \u001b[K\n",
      "print_rna_qc_metrics: calling get_elbow_knee_points \u001b[K\n",
      "[1] \"Returning sliced vectors\"\u001b[K\n",
      "[1] \"Returning sliced vectors\"\u001b[K\n",
      "null device \u001b[K\n",
      "          1 \u001b[K\n",
      "\u001b[K\n",
      "Finished scrna_plot_qc_metrics_prcs.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[32;1mCompleted at: 17-Nov-2023 01:54:08\n",
      "Duration    : 6m 54s\n",
      "CPU hours   : 5.5 (98.7% cached)\n",
      "Succeeded   : 2\n",
      "Cached      : 5\n",
      "\u001b[22;39m\u001b[K\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "!nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -with-tower -resume\n",
    "# -resume \n",
    "# 8af1ba30-6606-4dfd-a851-c6d0b74bdd43\n",
    "# 8af1ba30-6606-4dfd-a851-c6d0b74bdd43\n",
    "# 8af1ba30-6606-4dfd-a851-c6d0b74bdd43\n",
    "\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile docker_local -resume\n",
    "\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile  -with-tower\n",
    "\n",
    "# docker_local\n",
    "\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow.config -profile docker_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm_anvil.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_local.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_local.config -profile docker_profile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !ls /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/bin/scrna_plot_qc_metrics.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nov-17 01:47:07.554 [main] DEBUG nextflow.cli.Launcher - $> nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -with-tower -resume\n",
      "Nov-17 01:47:07.997 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.10.0\n",
      "Nov-17 01:47:08.044 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/home/eila/.nextflow/plugins; core-plugins: nf-amazon@2.1.4,nf-azure@1.3.2,nf-cloudcache@0.3.0,nf-codecommit@0.1.5,nf-console@1.0.6,nf-ga4gh@1.1.0,nf-google@1.8.3,nf-tower@1.6.3,nf-wave@1.0.0\n",
      "Nov-17 01:47:08.064 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Enabled plugins: []\n",
      "Nov-17 01:47:08.066 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Disabled plugins: []\n",
      "Nov-17 01:47:08.073 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode\n",
      "Nov-17 01:47:08.093 [main] INFO  org.pf4j.AbstractPluginManager - No plugins\n",
      "Nov-17 01:47:08.128 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/nextflow_vm.config\n",
      "Nov-17 01:47:08.129 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/nextflow_vm.config\n",
      "Nov-17 01:47:08.173 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Nov-17 01:47:11.370 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [docker_local, google_batch_profile, conda_profile]\n",
      "Nov-17 01:47:11.443 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 by global default\n",
      "Nov-17 01:47:11.468 [main] INFO  nextflow.cli.CmdRun - Launching `nf_wf_fg_singlecell_rna.nf` [jolly_hopper] DSL2 - revision: fe1c2abe14\n",
      "Nov-17 01:47:11.470 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[nf-google@1.8.3]\n",
      "Nov-17 01:47:11.472 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[nf-google@1.8.3, nf-tower@1.6.3, nf-wave@1.0.0]\n",
      "Nov-17 01:47:11.472 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-google version: 1.8.3\n",
      "Nov-17 01:47:11.498 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-google@1.8.3' resolved\n",
      "Nov-17 01:47:11.498 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-google@1.8.3'\n",
      "Nov-17 01:47:11.569 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-google@1.8.3\n",
      "Nov-17 01:47:11.569 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-tower version: 1.6.3\n",
      "Nov-17 01:47:11.571 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-tower@1.6.3' resolved\n",
      "Nov-17 01:47:11.572 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-tower@1.6.3'\n",
      "Nov-17 01:47:11.586 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-tower@1.6.3\n",
      "Nov-17 01:47:11.586 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-wave version: 1.0.0\n",
      "Nov-17 01:47:11.591 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-wave@1.0.0' resolved\n",
      "Nov-17 01:47:11.592 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-wave@1.0.0'\n",
      "Nov-17 01:47:11.603 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-wave@1.0.0\n",
      "Nov-17 01:47:11.626 [main] DEBUG n.secret.LocalSecretsProvider - Secrets store: /home/eila/.nextflow/secrets/store.json\n",
      "Nov-17 01:47:11.631 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@7f9e8421] - activable => nextflow.secret.LocalSecretsProvider@7f9e8421\n",
      "Nov-17 01:47:11.733 [main] DEBUG nextflow.Session - Session UUID: becedc50-945f-4f26-ab13-4ebd072d2c05\n",
      "Nov-17 01:47:11.733 [main] DEBUG nextflow.Session - Run name: jolly_hopper\n",
      "Nov-17 01:47:11.734 [main] DEBUG nextflow.Session - Executor pool size: 4\n",
      "Nov-17 01:47:12.124 [main] DEBUG nextflow.file.FilePorter - File porter settings maxRetries=3; maxTransfers=50; pollTimeout=null\n",
      "Nov-17 01:47:12.132 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=12; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false\n",
      "Nov-17 01:47:12.139 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/nextflow_vm.config\n",
      "Nov-17 01:47:12.140 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/nextflow_vm.config\n",
      "Nov-17 01:47:12.141 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Nov-17 01:47:14.293 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [docker_local, google_batch_profile, conda_profile]\n",
      "Nov-17 01:47:14.342 [main] DEBUG nextflow.cli.CmdRun - \n",
      "  Version: 23.10.0 build 5889\n",
      "  Created: 15-10-2023 15:07 UTC \n",
      "  System: Linux 5.10.0-26-cloud-amd64\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 11.0.13+7-b1751.21\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "  Process: 441483@eila-vm [10.128.0.63]\n",
      "  CPUs: 4 - Mem: 14.6 GB (408.1 MB) - Swap: 0 (0)\n",
      "Nov-17 01:47:14.425 [main] DEBUG nextflow.file.FileHelper - Can't check if specified path is NFS (1): gs://eila-nextflow-bucket/work/RNA\n",
      "\n",
      "Nov-17 01:47:14.425 [main] DEBUG nextflow.Session - Work-dir: gs://eila-nextflow-bucket/work/RNA [null]\n",
      "Nov-17 01:47:14.473 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[GoogleLifeSciencesExecutor, GoogleBatchExecutor]\n",
      "Nov-17 01:47:14.501 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory\n",
      "Nov-17 01:47:14.506 [main] DEBUG nextflow.Session - Observer factory: WaveFactory\n",
      "Nov-17 01:47:14.506 [main] DEBUG io.seqera.wave.plugin.WaveFactory - Detected Fusion enabled -- Enabling bundle project resources -- Disabling upload of remote bin directory\n",
      "Nov-17 01:47:14.542 [main] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Nov-17 01:47:14.558 [main] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null, dryRunMode:false)\n",
      "Nov-17 01:47:14.740 [main] DEBUG nextflow.Session - Observer factory: TowerFactory\n",
      "Nov-17 01:47:14.833 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory\n",
      "Nov-17 01:47:14.852 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 5; maxThreads: 1000\n",
      "Nov-17 01:47:15.013 [main] DEBUG nextflow.Session - Session start\n",
      "Nov-17 01:47:15.031 [main] DEBUG io.seqera.tower.plugin.TowerClient - Creating Tower observer -- endpoint=https://api.tower.nf; requestInterval=1s; aliveInterval=1m; maxRetries=5; backOffBase=3; backOffDelay=250\n",
      "Nov-17 01:47:16.267 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution\n",
      "Nov-17 01:47:17.194 [main] INFO  nextflow.script.BaseScript - null\n",
      "Nov-17 01:47:17.195 [main] INFO  nextflow.script.BaseScript - /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/rna_fastqs_files_vm-dogma-dig.csv\n",
      "Nov-17 01:47:17.409 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:seqspec_local` matches labels `seqspec_local` for process with name run_seqspec_modify_rna\n",
      "Nov-17 01:47:17.416 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Nov-17 01:47:17.416 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Nov-17 01:47:17.420 [main] DEBUG nextflow.executor.Executor - [warm up] executor > google-batch (fusion enabled)\n",
      "Nov-17 01:47:17.426 [main] DEBUG n.processor.TaskPollingMonitor - Creating task monitor for executor 'google-batch' > capacity: 1000; pollInterval: 10s; dumpInterval: 5m \n",
      "Nov-17 01:47:17.429 [main] DEBUG n.processor.TaskPollingMonitor - >>> barrier register (monitor: google-batch)\n",
      "Nov-17 01:47:17.435 [main] DEBUG nextflow.cloud.google.GoogleOpts - Google auth via application DEFAULT\n",
      "Nov-17 01:47:17.440 [main] DEBUG n.c.google.batch.GoogleBatchExecutor - [GOOGLE BATCH] Executor config=BatchConfig[googleOpts=GoogleOpts(projectId:hai-gcp-genomic, credsFile:null, location:us-central1, enableRequesterPaysBuckets:false, httpConnectTimeout:1m, httpReadTimeout:1m, credentials:UserCredentials{requestMetadata=null, temporaryAccess=null, clientId=764086051850-6qr4p6gpi6hn506pt8ejuq83di341hur.apps.googleusercontent.com, refreshToken=1//04-ZrKvCfXPa5CgYIARAAGAQSNwF-L9IrkFtcwyfUDAhTeAquw0t0utu5nMRfG-U9l5uflp_XJI0EMWlO9aPKmi0CxdbKiKRekQc, tokenServerUri=https://oauth2.googleapis.com/token, transportFactoryClassName=com.google.auth.oauth2.OAuth2Utils$DefaultHttpTransportFactory, quotaProjectId=hai-gcp-genomic})\n",
      "Nov-17 01:47:17.462 [main] DEBUG n.c.google.batch.client.BatchClient - [GOOGLE BATCH] Creating service client with config credentials\n",
      "Nov-17 01:47:18.154 [main] INFO  nextflow.script.BaseScript - after run_seqspec_modify_rna\n",
      "Nov-17 01:47:18.159 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:seqspec_local` matches labels `seqspec_local` for process with name run_seqspec_print\n",
      "Nov-17 01:47:18.161 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Nov-17 01:47:18.161 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Nov-17 01:47:18.166 [main] INFO  nextflow.script.BaseScript - after run_seqspec_print\n",
      "Nov-17 01:47:18.169 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:seqspec_local` matches labels `seqspec_local` for process with name run_seqspec_index_rna_kb\n",
      "Nov-17 01:47:18.171 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Nov-17 01:47:18.171 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Nov-17 01:47:18.175 [main] INFO  nextflow.script.BaseScript - after run_seqspec_index_rna_kb\n",
      "Nov-17 01:47:18.176 [main] INFO  nextflow.script.BaseScript - technology is $technology\n",
      "Nov-17 01:47:18.192 [main] INFO  nextflow.script.BaseScript - after genome_fasta_ch\n",
      "Nov-17 01:47:18.193 [main] INFO  nextflow.script.BaseScript - after genome_gtf_ch\n",
      "Nov-17 01:47:18.196 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:kb_ref` matches labels `kb_ref` for process with name run_kb_ref\n",
      "Nov-17 01:47:18.197 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Nov-17 01:47:18.197 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Nov-17 01:47:18.202 [main] INFO  nextflow.script.BaseScript - after run_kb_ref\n",
      "Nov-17 01:47:18.203 [main] INFO  nextflow.script.BaseScript - after get organism\n",
      "Nov-17 01:47:18.207 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:kb_count` matches labels `kb_count` for process with name run_kb_count\n",
      "Nov-17 01:47:18.209 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Nov-17 01:47:18.209 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Nov-17 01:47:18.215 [main] INFO  nextflow.script.BaseScript - after run_kb_count\n",
      "Nov-17 01:47:18.220 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:scrna_qc_calculate` matches labels `scrna_qc_calculate` for process with name scrna_calculate_qc_metrics\n",
      "Nov-17 01:47:18.222 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Nov-17 01:47:18.222 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Nov-17 01:47:18.228 [main] INFO  nextflow.script.BaseScript - scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\n",
      "Nov-17 01:47:18.229 [main] INFO  nextflow.script.BaseScript - scrna_qc_metrics_tsv is: DataflowBroadcast around DataflowStream[?]\n",
      "Nov-17 01:47:18.230 [main] INFO  nextflow.script.BaseScript - sc_rna_qc_plot_script is: scrna_plot_qc_metrics.R\n",
      "Nov-17 01:47:18.231 [main] INFO  nextflow.script.BaseScript - sc_rna_qc_plot_helper_script is: barcode_rank_functions.R\n",
      "Nov-17 01:47:18.236 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:scrna_qc_plot` matches labels `scrna_qc_plot` for process with name scrna_plot_qc_metrics_prcs\n",
      "Nov-17 01:47:18.240 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Nov-17 01:47:18.240 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Nov-17 01:47:18.252 [main] DEBUG nextflow.Session - Workflow process names [dsl2]: run_kb_count, run_seqspec_modify_rna, run_seqspec_print, run_kb_ref, run_synpase_upload, run_download_kb_idx, scrna_plot_qc_metrics_prcs, run_seqspec_check, run_synpase_download, run_seqspec_index_rna_kb, run_seqspec_test, scrna_calculate_qc_metrics, run_seqspec_modify_atac, run_downloadFiles\n",
      "Nov-17 01:47:18.509 [main] INFO  io.seqera.tower.plugin.TowerClient - Monitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/AFnwNuAoOlZcG\n",
      "Nov-17 01:47:18.510 [main] DEBUG nextflow.Session - Igniting dataflow network (12)\n",
      "Nov-17 01:47:18.522 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_seqspec_modify_rna\n",
      "Nov-17 01:47:18.524 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_seqspec_print\n",
      "Nov-17 01:47:18.526 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_seqspec_index_rna_kb\n",
      "Nov-17 01:47:18.526 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_kb_ref\n",
      "Nov-17 01:47:18.527 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_kb_count\n",
      "Nov-17 01:47:18.531 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > scrna_calculate_qc_metrics\n",
      "Nov-17 01:47:18.540 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > scrna_plot_qc_metrics_prcs\n",
      "Nov-17 01:47:18.548 [main] DEBUG nextflow.script.ScriptRunner - Parsed script files:\n",
      "  Script_8c658adfb468bfd1: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/./../../nf_processes/nf_prcs_seqspec_utils.nf\n",
      "  Script_6f4853daaf022cb7: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/./../../nf_processes/nf_prcs_synapse_utils.nf\n",
      "  Script_b097b54f100d695d: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/nf_wf_fg_singlecell_rna.nf\n",
      "  Script_e6cddbc2c25317da: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/./../../nf_processes/nf_prcs_rna_qc.nf\n",
      "  Script_9d4639c8cd883c70: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/./../../nf_processes/nf_prcs_download_url_files.nf\n",
      "  Script_88de0ef9af95954d: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/./../../nf_processes/nf_prcs_kallisto_bustools.nf\n",
      "  Script_905602450a702dad: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/./../../nf_processes/nf_prcs_rna_qc_plots.nf\n",
      "Nov-17 01:47:18.548 [main] DEBUG nextflow.script.ScriptRunner - > Awaiting termination \n",
      "Nov-17 01:47:18.548 [main] DEBUG nextflow.Session - Session await\n",
      "Nov-17 01:47:21.454 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Local cache found for foreign file http://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz at gs://eila-nextflow-bucket/work/RNA/stage-becedc50-945f-4f26-ab13-4ebd072d2c05/70/4e80d977dc0161ca71234c7b24bd5a/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\n",
      "Nov-17 01:47:21.516 [FileTransfer-2] DEBUG nextflow.file.FilePorter - Local cache found for foreign file http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz at gs://eila-nextflow-bucket/work/RNA/stage-becedc50-945f-4f26-ab13-4ebd072d2c05/db/6a8dad39a29382bb0b306219fec0c4/Homo_sapiens.GRCh38.109.gtf.gz\n",
      "Nov-17 01:47:21.536 [Actor Thread 7] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Nov-17 01:47:21.537 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null, dryRunMode:false)\n",
      "Nov-17 01:47:21.549 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave assets bundle bin resources: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/bin\n",
      "Nov-17 01:47:21.806 [FileTransfer-6] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/nf_spec_dogmaseq_dig.yaml at gs://eila-nextflow-bucket/work/RNA/stage-becedc50-945f-4f26-ab13-4ebd072d2c05/ec/da44e36f187b61af8a7154ad890c50/nf_spec_dogmaseq_dig.yaml\n",
      "Nov-17 01:47:21.816 [FileTransfer-7] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/RNA-737K-arc-v1.txt at gs://eila-nextflow-bucket/work/RNA/stage-becedc50-945f-4f26-ab13-4ebd072d2c05/48/9fb1bf6459e32b854a354795cc23ed/RNA-737K-arc-v1.txt\n",
      "Nov-17 01:47:21.882 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave request container config: https://fusionfs.seqera.io/releases/v2.2-amd64.json\n",
      "Nov-17 01:47:21.894 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/fastqs/RNA/SRR18677638_1.fastq.gz at gs://eila-nextflow-bucket/work/RNA/stage-becedc50-945f-4f26-ab13-4ebd072d2c05/4a/e36d3c2995bc744385c7debd26cf14/SRR18677638_1.fastq.gz\n",
      "Nov-17 01:47:21.998 [FileTransfer-4] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/fastqs/RNA/SRR18677638_2.fastq.gz at gs://eila-nextflow-bucket/work/RNA/stage-becedc50-945f-4f26-ab13-4ebd072d2c05/1d/89d2ca2e8fa5a306fea12ad7f0110d/SRR18677638_2.fastq.gz\n",
      "Nov-17 01:47:22.016 [FileTransfer-5] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/na.na at gs://eila-nextflow-bucket/work/RNA/stage-becedc50-945f-4f26-ab13-4ebd072d2c05/54/2e3383df0222e80089b790eabd0ced/na.na\n",
      "Nov-17 01:47:22.098 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave container config response: [200] {\n",
      "  \"entrypoint\": [ \"/usr/bin/fusion\" ],\n",
      "  \"layers\": [\n",
      "    {\n",
      "      \"location\": \"https://fusionfs.seqera.io/releases/pkg/2/2/9/fusion-amd64.tar.gz\",\n",
      "      \"gzipDigest\": \"sha256:deedfe8b52c6230681074c3f97f3a87e5af4983ccded1174e8a556de5496d0e3\",\n",
      "      \"gzipSize\": 30266040,\n",
      "      \"tarDigest\": \"sha256:cfaf78238ffcdd1b4e050750af1c219dbcb7a59f8e0a1e88a2ff932b648fa252\",\n",
      "      \"skipHashing\": true\n",
      "    }\n",
      "  ]\n",
      "}\n",
      "\n",
      "Nov-17 01:47:22.205 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/atomic_update:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+1de3fb...; tarDigest=sha256:d16186cd9484b7018d884e2981502445eff14e549ae832b24cefc5677c787dfc; gzipDigest=sha256:940f846b17e7b2b18a7c5563ec941cdee33ed5aecb7a6d1eabcff5a788fafd7b; gzipSize=8198], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/9/fusion-amd64.tar.gz; tarDigest=sha256:cfaf78238ffcdd1b4e050750af1c219dbcb7a59f8e0a1e88a2ff932b648fa252; gzipDigest=sha256:deedfe8b52c6230681074c3f97f3a87e5af4983ccded1174e8a556de5496d0e3; gzipSize=30266040]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2023-11-17T01:47:22.202833Z, fingerprint:3c349660dfed38d477c7f11f996a4746, freeze:false, format:null, dryRun:false)\n",
      "Nov-17 01:47:22.205 [Actor Thread 6] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/seqspec_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+1de3fb...; tarDigest=sha256:d16186cd9484b7018d884e2981502445eff14e549ae832b24cefc5677c787dfc; gzipDigest=sha256:940f846b17e7b2b18a7c5563ec941cdee33ed5aecb7a6d1eabcff5a788fafd7b; gzipSize=8198], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/9/fusion-amd64.tar.gz; tarDigest=sha256:cfaf78238ffcdd1b4e050750af1c219dbcb7a59f8e0a1e88a2ff932b648fa252; gzipDigest=sha256:deedfe8b52c6230681074c3f97f3a87e5af4983ccded1174e8a556de5496d0e3; gzipSize=30266040]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2023-11-17T01:47:22.202833Z, fingerprint:abe7fe2e8e6a282e21a3b2ab91c81ec7, freeze:false, format:null, dryRun:false)\n",
      "Nov-17 01:47:22.817 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"a7ff7cb9e685\",\"targetImage\":\"wave.seqera.io/wt/a7ff7cb9e685/eilalan/atomic_update:latest\",\"expiration\":\"2023-11-18T13:47:22.763002966Z\",\"containerImage\":\"docker.io/eilalan/atomic_update:latest\"}\n",
      "Nov-17 01:47:22.817 [Actor Thread 6] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"763cfa8b4cee\",\"targetImage\":\"wave.seqera.io/wt/763cfa8b4cee/eilalan/seqspec_env:latest\",\"expiration\":\"2023-11-18T13:47:22.762824279Z\",\"containerImage\":\"docker.io/eilalan/seqspec_env:latest\"}\n",
      "Nov-17 01:47:23.960 [Actor Thread 7] INFO  nextflow.processor.TaskProcessor - [04/b164fb] Cached process > run_kb_ref\n",
      "Nov-17 01:47:24.045 [Actor Thread 6] INFO  nextflow.processor.TaskProcessor - [39/e50e47] Cached process > run_seqspec_modify_rna (1)\n",
      "Nov-17 01:47:24.862 [Actor Thread 10] INFO  nextflow.processor.TaskProcessor - [77/6b6ef6] Cached process > run_seqspec_print (1)\n",
      "Nov-17 01:47:25.268 [Actor Thread 9] INFO  nextflow.processor.TaskProcessor - [b2/06d00c] Cached process > run_seqspec_index_rna_kb (1)\n",
      "Nov-17 01:47:27.012 [Actor Thread 10] INFO  nextflow.processor.TaskProcessor - [41/090813] Cached process > run_kb_count (1)\n",
      "Nov-17 01:47:27.031 [Actor Thread 9] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/rna_qc_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+1de3fb...; tarDigest=sha256:d16186cd9484b7018d884e2981502445eff14e549ae832b24cefc5677c787dfc; gzipDigest=sha256:940f846b17e7b2b18a7c5563ec941cdee33ed5aecb7a6d1eabcff5a788fafd7b; gzipSize=8198], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/9/fusion-amd64.tar.gz; tarDigest=sha256:cfaf78238ffcdd1b4e050750af1c219dbcb7a59f8e0a1e88a2ff932b648fa252; gzipDigest=sha256:deedfe8b52c6230681074c3f97f3a87e5af4983ccded1174e8a556de5496d0e3; gzipSize=30266040]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2023-11-17T01:47:27.030485Z, fingerprint:54edf2a6b09f4124d278f280ba5c2c3c, freeze:false, format:null, dryRun:false)\n",
      "Nov-17 01:47:27.152 [Actor Thread 9] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"53da0e095dd7\",\"targetImage\":\"wave.seqera.io/wt/53da0e095dd7/eilalan/rna_qc_env:latest\",\"expiration\":\"2023-11-18T13:47:27.099314967Z\",\"containerImage\":\"docker.io/eilalan/rna_qc_env:latest\"}\n",
      "Nov-17 01:47:28.642 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `scrna_calculate_qc_metrics (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Nov-17 01:47:30.334 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `scrna_calculate_qc_metrics (1)` submitted > job=nf-418fd5ce-1700185647903; uid=nf-418fd5ce-170018-768e72de-09ed-49350; work-dir=gs://eila-nextflow-bucket/work/RNA/41/8fd5ce3b282820cda26d036fa83f7d\n",
      "Nov-17 01:47:30.334 [Task submitter] INFO  nextflow.Session - [41/8fd5ce] Submitted process > scrna_calculate_qc_metrics (1)\n",
      "Nov-17 01:50:57.473 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `scrna_calculate_qc_metrics (1)` - terminated job=nf-418fd5ce-1700185647903; state=SUCCEEDED\n",
      "Nov-17 01:50:57.802 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 6; name: scrna_calculate_qc_metrics (1); status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/RNA/41/8fd5ce3b282820cda26d036fa83f7d]\n",
      "Nov-17 01:50:58.040 [Actor Thread 1] DEBUG io.seqera.wave.plugin.WaveClient - Wave request container config: https://fusionfs.seqera.io/releases/v2.2-amd64.json\n",
      "Nov-17 01:50:58.098 [Actor Thread 1] DEBUG io.seqera.wave.plugin.WaveClient - Wave container config response: [200] {\n",
      "  \"entrypoint\": [ \"/usr/bin/fusion\" ],\n",
      "  \"layers\": [\n",
      "    {\n",
      "      \"location\": \"https://fusionfs.seqera.io/releases/pkg/2/2/9/fusion-amd64.tar.gz\",\n",
      "      \"gzipDigest\": \"sha256:deedfe8b52c6230681074c3f97f3a87e5af4983ccded1174e8a556de5496d0e3\",\n",
      "      \"gzipSize\": 30266040,\n",
      "      \"tarDigest\": \"sha256:cfaf78238ffcdd1b4e050750af1c219dbcb7a59f8e0a1e88a2ff932b648fa252\",\n",
      "      \"skipHashing\": true\n",
      "    }\n",
      "  ]\n",
      "}\n",
      "\n",
      "Nov-17 01:50:59.106 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `scrna_plot_qc_metrics_prcs (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Nov-17 01:50:59.411 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `scrna_plot_qc_metrics_prcs (1)` submitted > job=nf-2a1154df-1700185858639; uid=nf-2a1154df-170018-f0ac3599-6131-4e030; work-dir=gs://eila-nextflow-bucket/work/RNA/2a/1154df225ae96d7d4b8c2c6c6ddad0\n",
      "Nov-17 01:50:59.411 [Task submitter] INFO  nextflow.Session - [2a/1154df] Submitted process > scrna_plot_qc_metrics_prcs (1)\n",
      "Nov-17 01:52:27.445 [Task monitor] DEBUG n.processor.TaskPollingMonitor - !! executor google-batch > tasks to be completed: 1 -- submitted tasks are shown below\n",
      "~> TaskHandler[id: 7; name: scrna_plot_qc_metrics_prcs (1); status: SUBMITTED; exit: -; error: -; workDir: gs://eila-nextflow-bucket/work/RNA/2a/1154df225ae96d7d4b8c2c6c6ddad0]\n",
      "Nov-17 01:54:07.535 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `scrna_plot_qc_metrics_prcs (1)` - terminated job=nf-2a1154df-1700185858639; state=SUCCEEDED\n",
      "Nov-17 01:54:07.886 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 7; name: scrna_plot_qc_metrics_prcs (1); status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/RNA/2a/1154df225ae96d7d4b8c2c6c6ddad0]\n",
      "Nov-17 01:54:08.523 [main] DEBUG nextflow.Session - Session await > all processes finished\n",
      "Nov-17 01:54:08.776 [Task monitor] DEBUG n.processor.TaskPollingMonitor - <<< barrier arrives (monitor: google-batch) - terminating tasks monitor poll loop\n",
      "Nov-17 01:54:08.776 [main] DEBUG nextflow.Session - Session await > all barriers passed\n",
      "Nov-17 01:54:08.784 [main] DEBUG n.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=2; failedCount=0; ignoredCount=0; cachedCount=5; pendingCount=0; submittedCount=0; runningCount=0; retriesCount=0; abortedCount=0; succeedDuration=4m 23s; failedDuration=0ms; cachedDuration=5h 26m 40s;loadCpus=0; loadMemory=0; peakRunning=1; peakCpus=4; peakMemory=0; ]\n",
      "Nov-17 01:54:09.473 [main] DEBUG nextflow.cache.CacheDB - Closing CacheDB done\n",
      "Nov-17 01:54:09.474 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-wave@1.0.0'\n",
      "Nov-17 01:54:09.474 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-wave\n",
      "Nov-17 01:54:09.474 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-tower@1.6.3'\n",
      "Nov-17 01:54:09.474 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-tower\n",
      "Nov-17 01:54:09.474 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-google@1.8.3'\n",
      "Nov-17 01:54:09.474 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-google\n",
      "Nov-17 01:54:09.508 [main] DEBUG nextflow.util.ThreadPoolManager - Thread pool 'FileTransfer' shutdown completed (hard=false)\n",
      "Nov-17 01:54:09.509 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye\n"
     ]
    }
   ],
   "source": [
    "!cat .nextflow.log "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-11-17 01:54:10.243202\n"
     ]
    }
   ],
   "source": [
    "import datetime\n",
    "now = datetime.datetime.now()\n",
    "print(now)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -with-tower -resume \n",
    "# 8af1ba30-6606-4dfd-a851-c6d0b74bdd43\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -resume 886dc4ef-ff60-46d0-aecb-8f1ab56296a3\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -resume\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile docker_local\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -resume\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -resume\n",
    "\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile docker_local -resume\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -resume\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow.config -profile docker_profile -resume\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -resume\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_local.config -profile google_batch_profile -resume\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -resume\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow.config -profile docker_profile -resume\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow.config -profile google_batch_profile -resume"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !nextflow log\n",
    "# use the session id with the resume"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !cat .nextflow.log "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-11-17 01:54:10.276189\n"
     ]
    }
   ],
   "source": [
    "import datetime\n",
    "now = datetime.datetime.now()\n",
    "print(now)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
