{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 869,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell'"
      ]
     },
     "execution_count": 869,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 870,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # # todo: synapse_config for the nf-core\n",
    "# # docker {\n",
    "# #   process.container = 'eilalan/env_synapse:latest'\n",
    "# #   docker.enabled = true\n",
    "# #   // Include ENV_SYNAPSE_TOKEN in envWhitelist\n",
    "# #   envWhitelist=['ENV_SYNAPSE_TOKEN']\n",
    "\n",
    "# # }\n",
    "# process {\n",
    "#   withName:run_synpase_download {\n",
    "#     container = 'eilalan/env_synapse:latest'\n",
    "#     executor = 'local'\n",
    "#   }\n",
    "# }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 871,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Overwriting nextflow.config\n"
     ]
    }
   ],
   "source": [
    "%%writefile nextflow.config\n",
    "\n",
    "includeConfig 'conf/conda.config'\n",
    "\n",
    "\n",
    "manifest {\n",
    "  description = 'A nexflow pipeline for fg single cell' \n",
    "  mainScript = 'nf_wf_fg_singlecell.nf'\n",
    "}\n",
    "\n",
    "profiles {\n",
    "  conda_profile {\n",
    "    includeConfig 'conf/conda.config'\n",
    "    conda.enabled = true\n",
    "    executor = 'local'\n",
    "    }\n",
    "\n",
    "  conda_docker_profile {\n",
    "    includeConfig 'conf/conda_docker.config'\n",
    "    executor = 'local'\n",
    "    }\n",
    "\n",
    "  docker_profile {\n",
    "    includeConfig 'conf/docker.config'\n",
    "    docker.enabled = true\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN']\n",
    "    executor = 'local'\n",
    "  }\n",
    "}\n",
    "\n",
    "// Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "params.ORGANISM = 'human'\n",
    "params.FASTQS_SPEC_CH = '/Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/rna_fastqs_files_local.csv'\n",
    "params.GENOME_FASTA='http://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz'\n",
    "params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "params.ENV_SYNAPSE_TOKEN = \"nextflow secrets get SYNAPSE\".execute().text.trim()\n",
    "params.SYNAPSE_CONFIG = '/Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/synapse.config'\n",
    "params.SYNAPSE_IDS = '/Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/synapse_ids.csv'\n",
    "params.OUTDIR='./igvf-out'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 872,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 873,
   "metadata": {},
   "outputs": [],
   "source": [
    "!rm -r /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell/work"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 874,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rm: cannot remove '.nextflow': Is a directory\n"
     ]
    }
   ],
   "source": [
    "!rm .nextflow*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 875,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nextflow.config\n"
     ]
    }
   ],
   "source": [
    "!ls nextflow.config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 888,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# !nextflow run nf_wf_fg_singlecell.nf -c nextflow.config -profile docker_profile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 911,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.04.4\n",
      "Launching `nf_wf_fg_singlecell.nf` [berserk_yalow] DSL2 - revision: 6ae376f630\n",
      "eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIl0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTY5NzgxODkxNiwiaWF0IjoxNjk3ODE4OTE2LCJqdGkiOiI0MTIzIiwic3ViIjoiMzQ3NDU3MSJ9.n0cugYMAVzYZ8mMMc7CjcqqEV7FB-K-oKUDT45SvM6KmbCpb_v3lZ2gg3xXi1tubWH0__wpnbJ7cYrGEtOyt4vXAPkMiEtZFGqS0MFKp_h-mHHH4pnACSQQYEy1zdyqFn9WRE7ipjWg4yYSbq84N7J3gzi8VKmiOAmuIGKaJaS0snkUjtJEVQH-0vZqWs1nKgVRWIZv31Bo8vuDzWpBrXD-zRT_uQW9l_I9fU50o1AqyFhNe_gbty_Z6imb8VT33wMXiDgJh15lG_M70PXU15v3egDXLgw1dGj32kj1nKanfqBVqbMSzgzPAG037p3YH_Q556BLceFBBb5Jaej5JVA\n",
      "/Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/fastqs_files_local.csv\n",
      "[fc/a72ca1] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[89/83f3ac] process > run_seqspec_check (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[57/bec9d4] process > run_seqspec_modify_rna (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[ac/65affa] process > run_seqspec_index_rna_kb (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[6f/f231ec] process > run_download_kb_idx          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_kb_count                 -\u001b[K\n",
      "DataflowVariable(value=null)\u001b[K\n",
      "DataflowVariable(value=null)\u001b[K\n",
      "DataflowVariable(value=null)\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "ProcessDef[process run_kb_count]\u001b[K\n",
      "start run_seqspec_modify_rna\u001b[K\n",
      "rna_R1_SRR18677638.fastq.gz\u001b[K\n",
      "rna_R2_SRR18677638.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "rna_R1_SRR18677638.fastq.gz\u001b[K\n",
      "rna_R2_SRR18677638.fastq.gz\u001b[K\n",
      "finished seqspec modify\u001b[K\n",
      "\u001b[K\n",
      "kb ref -d human  -i index.idx -g t2g.txt --verbose\u001b[K\n",
      "total 3090812\u001b[K\n",
      "-rw-r--r-- 1 501 staff 3145942371 Oct 25  2019 index.idx\u001b[K\n",
      "-rw-r--r-- 1 501 staff   10036885 Oct 25  2019 t2g.txt\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_check\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec check spec_yaml\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec print nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "\u001b[K\n",
      "spec_yaml is nf_seqspec_modify.yaml\u001b[K\n",
      "RNA_R1_fastq_gz is rna_R1_SRR18677638.fastq.gz\u001b[K\n",
      "RNA_R2_fastq_gz is rna_R2_SRR18677638.fastq.gz\u001b[K\n",
      "0,0,16:0,16,28:1,0,102\u001b[K\n",
      "finished run_seqspec_index_rna\u001b[K\n",
      "\u001b[K\n",
      "\u001b[40A\n",
      "executor >  local (1)\u001b[K\n",
      "[fc/a72ca1] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[89/83f3ac] process > run_seqspec_check (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[57/bec9d4] process > run_seqspec_modify_rna (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[ac/65affa] process > run_seqspec_index_rna_kb (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[6f/f231ec] process > run_download_kb_idx          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[ed/d084fa] process > run_kb_count (1)             [  0%] 0 of 1\u001b[K\n",
      "DataflowVariable(value=null)\u001b[K\n",
      "DataflowVariable(value=null)\u001b[K\n",
      "DataflowVariable(value=null)\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "ProcessDef[process run_kb_count]\u001b[K\n",
      "start run_seqspec_modify_rna\u001b[K\n",
      "rna_R1_SRR18677638.fastq.gz\u001b[K\n",
      "rna_R2_SRR18677638.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "rna_R1_SRR18677638.fastq.gz\u001b[K\n",
      "rna_R2_SRR18677638.fastq.gz\u001b[K\n",
      "finished seqspec modify\u001b[K\n",
      "\u001b[K\n",
      "kb ref -d human  -i index.idx -g t2g.txt --verbose\u001b[K\n",
      "total 3090812\u001b[K\n",
      "-rw-r--r-- 1 501 staff 3145942371 Oct 25  2019 index.idx\u001b[K\n",
      "-rw-r--r-- 1 501 staff   10036885 Oct 25  2019 t2g.txt\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_check\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec check spec_yaml\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec print nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "\u001b[K\n",
      "spec_yaml is nf_seqspec_modify.yaml\u001b[K\n",
      "RNA_R1_fastq_gz is rna_R1_SRR18677638.fastq.gz\u001b[K\n",
      "RNA_R2_fastq_gz is rna_R2_SRR18677638.fastq.gz\u001b[K\n",
      "0,0,16:0,16,28:1,0,102\u001b[K\n",
      "finished run_seqspec_index_rna\u001b[K\n",
      "\u001b[K\n",
      "\u001b[41A\n",
      "executor >  local (1)\u001b[K\n",
      "[fc/a72ca1] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[89/83f3ac] process > run_seqspec_check (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[57/bec9d4] process > run_seqspec_modify_rna (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[ac/65affa] process > run_seqspec_index_rna_kb (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[6f/f231ec] process > run_download_kb_idx          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[ed/d084fa] process > run_kb_count (1)             [100%] 1 of 1 ✔\u001b[K\n",
      "start run_kb_count !!!\u001b[K\n",
      "index.idx\u001b[K\n",
      "t2g.txt\u001b[K\n",
      "rna_R1_SRR18677638.fastq.gz\u001b[K\n",
      "rna_R2_SRR18677638.fastq.gz\u001b[K\n",
      "seqspec.index.out\u001b[K\n",
      "0,0,16:0,16,28:1,0,102\u001b[K\n",
      "kb count -i index.idx -g t2g.txt -x 0,0,16:0,16,28:1,0,102 -o out --h5ad -t 8 rna_R1_SRR18677638.fastq.gz rna_R2_SRR18677638.fastq.gz\u001b[K\n",
      "Homo_sapiens.GRCh38.109.gtf.gz\u001b[K\n",
      "index.idx\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "out\u001b[K\n",
      "rna_R1_SRR18677638.fastq.gz\u001b[K\n",
      "rna_R2_SRR18677638.fastq.gz\u001b[K\n",
      "seqspec.index.out\u001b[K\n",
      "t2g.txt\u001b[K\n",
      "counts_unfiltered\u001b[K\n",
      "inspect.json\u001b[K\n",
      "kb_info.json\u001b[K\n",
      "matrix.ec\u001b[K\n",
      "output.bus\u001b[K\n",
      "output.unfiltered.bus\u001b[K\n",
      "run_info.json\u001b[K\n",
      "transcripts.txt\u001b[K\n",
      "whitelist.txt\u001b[K\n",
      "finished run_kb_count\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[32;1mCompleted at: 25-Oct-2023 13:35:31\n",
      "Duration    : 2m 34s\n",
      "CPU hours   : 2.4 (86% cached)\n",
      "Succeeded   : 1\n",
      "Cached      : 5\n",
      "\u001b[22;39m\u001b[K\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow run nf_wf_fg_singlecell.nf -c nextflow.config -profile docker_profile -resume"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 810,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "atac_R1_SRR18677642.fastq.gz\t protein_R2_SRR18677644.fastq.gz\n",
      "atac_R2_SRR18677642.fastq.gz\t tag_R1_SRR18677640.fastq.gz\n",
      "atac_R3_SRR18677642.fastq.gz\t tag_R2_SRR18677640.fastq.gz\n",
      "protein_R1_SRR18677644.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "# !ls /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/seqspec/specs/dogmaseq-dig/fastqs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 672,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !nextflow run /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_processes/nf_prcs_print.nf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 673,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !ls /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_processes/nf_prcs_print.nf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 776,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !cat .command.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 685,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !nextflow run nf_wf_fg_singlecell.nf -c nextflow.config -with-docker eilalan/env_synapse:latest"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
