{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# start docker deamon in terminal\n",
    "# !sudo docker login\n",
    "# OR - becuase the jupyther with allow root\n",
    "# !docker login"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO:\n",
    "\n",
    "when running with cloud. \n",
    "if you dont need to keep the execution logs, change the cloud settings to keep it for a short while"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# # in the command line. same env\n",
    "# export GOOGLE_APPLICATION_CREDENTIALS=\"/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/hai-gcp-genomic-beb9207a40b8.json.json\"\n",
    "# gcloud auth activate-service-account eila-batch@hai-gcp-genomic.iam.gserviceaccount.com --key-file=$GOOGLE_APPLICATION_CREDENTIALS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  Credentialed Accounts\n",
      "ACTIVE  ACCOUNT\n",
      "        886987709752-compute@developer.gserviceaccount.com\n",
      "*       eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\n",
      "\n",
      "To set the active account, run:\n",
      "    $ gcloud config set account `ACCOUNT`\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!gcloud auth list\n",
    "# gcloud config set account\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "# os.environ[\"TOWER_ACCESS_TOKEN\"] = 'XXX'\n",
    "os.environ[\"TOWER_ACCESS_TOKEN\"] = 'eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "os.environ[\"run_seqspec_modify_atac\"] = 'eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "os.environ[\"TOWER_WORKSPACE_ID\"]='247507587970818'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz\n",
      "247507587970818\n"
     ]
    }
   ],
   "source": [
    "!echo $TOWER_ACCESS_TOKEN\n",
    "!echo $TOWER_WORKSPACE_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin/atac_barcode_and_subpool.sh\n"
     ]
    }
   ],
   "source": [
    "!ls /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin/atac_barcode_and_subpool.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!chmod +x /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin/atac_barcode_and_subpool.sh\n",
    "!chmod +x /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin/run_tabix.sh\n",
    "!chmod +x /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin/create_sample_filtered_fragment_file.sh\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# in the command line. based on:\n",
    "# https://nextflow.io/blog/2023/nextflow-with-gbatch.html\n",
    "# gcloud services enable batch.googleapis.com compute.googleapis.com logging.googleapis.com\n",
    "# gcloud iam service-accounts create eila-batch\n",
    "# gcloud iam service-accounts list \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# gcloud projects add-iam-policy-binding terra-addd4c49 \\\n",
    "# --member=\"serviceAccount:pet-26869426946125b815f6b@terra-addd4c49.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/iam.serviceAccountUser\"\n",
    "\n",
    "# $ gcloud projects add-iam-policy-binding terra-addd4c49 \\\n",
    "# --member=\"serviceAccount:pet-26869426946125b815f6b@terra-addd4c49.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/batch.jobsEditor\"\n",
    "\n",
    "# $ gcloud projects add-iam-policy-binding terra-addd4c49 \\\n",
    "# --member=\"serviceAccount:pet-26869426946125b815f6b@terra-addd4c49.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/logging.viewer\"\n",
    "\n",
    "# $ gcloud projects add-iam-policy-binding terra-addd4c49 \\\n",
    "# --member=\"serviceAccount:pet-26869426946125b815f6b@terra-addd4c49.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/storage.admin\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/iam.serviceAccountUser\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/batch.jobsEditor\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/logging.viewer\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !nextflow -self-update"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Version: 23.10.0 build 5889\n",
      "  Created: 15-10-2023 15:07 UTC \n",
      "  System: Linux 5.10.0-26-cloud-amd64\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 11.0.13+7-b1751.21\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "docker\t\t\t\t   nf_data     nf_processes\n",
      "hai-gcp-genomic-beb9207a40b8.json  nf_modules  nf_workflow\n"
     ]
    }
   ],
   "source": [
    "!ls /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/RNA-737K-arc-v1.txt\n"
     ]
    }
   ],
   "source": [
    "!ls /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/RNA-737K-arc-v1.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac'"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Overwriting nextflow_local.config\n"
     ]
    }
   ],
   "source": [
    "%%writefile nextflow_local.config\n",
    "\n",
    "includeConfig 'conf/conda.config'\n",
    "\n",
    "\n",
    "manifest {\n",
    "  description = 'A nexflow pipeline for fg single cell atac' \n",
    "  mainScript = 'nf_wf_fg_singlecell_atac.nf'\n",
    "}\n",
    "profiles {\n",
    "  conda_profile {\n",
    "    includeConfig 'conf/conda.config'\n",
    "    conda.enabled = true\n",
    "    executor = 'local'\n",
    "    }\n",
    "\n",
    "  conda_docker_profile {\n",
    "    includeConfig 'conf/conda_docker.config'\n",
    "    executor = 'local'\n",
    "    }\n",
    "\n",
    "  docker_profile {\n",
    "    includeConfig 'conf/docker.config'\n",
    "    docker.enabled = true\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN']\n",
    "    executor = 'local'\n",
    "  }\n",
    "  google_batch_profile {\n",
    "    workDir = 'gs://eila-nextflow-bucket/work/ATAC'\n",
    "    includeConfig 'conf/docker-google-batch.config'\n",
    "    docker.enabled = true\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN']\n",
    "    executor = 'google-batch'\n",
    "    google.project = 'hai-gcp-genomic'\n",
    "    google.location = 'us-central1'\n",
    "    google.batch.spot = false\n",
    "    google.batch.serviceAccountEmail='pet-26869426946125b815f6b@terra-addd4c49.iam.gserviceaccount.com'\n",
    "  }\n",
    "    \n",
    "}\n",
    "params.FASTQS_SPEC_CH = '/Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/atac_fastqs_files_local.csv'\n",
    "params.CHROMAP_IDX='/Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index'\n",
    "\n",
    "params.GENOME_FASTA='http://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz'\n",
    "params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "\n",
    "params.CHROMAP_TRIM_ADAPTERS = false"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# %%writefile nextflow_vm.config\n",
    "\n",
    "# includeConfig 'conf/conda.config'\n",
    "\n",
    "\n",
    "# manifest {\n",
    "#   description = 'A nexflow pipeline for fg single cell atac' \n",
    "#   mainScript = 'nf_wf_fg_singlecell_atac.nf'\n",
    "# }\n",
    "\n",
    "# profiles {\n",
    "#   conda_profile {\n",
    "#     includeConfig 'conf/conda.config'\n",
    "#     conda.enabled = true\n",
    "#     executor = 'local'\n",
    "#     }\n",
    "\n",
    "#   conda_docker_profile {\n",
    "#     includeConfig 'conf/conda_docker.config'\n",
    "#     executor = 'local'\n",
    "#     }\n",
    "\n",
    "#   docker_profile {\n",
    "#     includeConfig 'conf/docker.config'\n",
    "#     docker.enabled = true\n",
    "#     docker.envWhitelist=['ENV_SYNAPSE_TOKEN']\n",
    "#     executor = 'local'\n",
    "#   }\n",
    "#   google_batch_profile {\n",
    "#     workDir = 'gs://eila-nextflow-bucket/work/ATAC'\n",
    "#     wave.enabled = true\n",
    "#     fusion.enabled = true\n",
    "#     includeConfig 'conf/docker-google-batch.config'\n",
    "#     docker.enabled = true\n",
    "#     docker.envWhitelist=['ENV_SYNAPSE_TOKEN']\n",
    "#     executor = 'google-batch'\n",
    "#     google.project = 'hai-gcp-genomic'\n",
    "#     google.location = 'us-central1'\n",
    "#     google.batch.spot = false\n",
    "#     google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "#   }\n",
    "    \n",
    "# }\n",
    "\n",
    "# params.FASTQS_SPEC_CH = '/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/atac_fastqs_files_vm.csv'\n",
    "# params.CHROMAP_IDX='/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index'\n",
    "\n",
    "# params.GENOME_FASTA='http://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz'\n",
    "# params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Overwriting nextflow_vm.config\n"
     ]
    }
   ],
   "source": [
    "%%writefile nextflow_vm.config\n",
    "\n",
    "includeConfig 'conf/conda.config'\n",
    "\n",
    "manifest {\n",
    "  description = 'A nexflow pipeline for fg single cell atac' \n",
    "  mainScript = 'nf_wf_fg_singlecell_atac.nf'\n",
    "}\n",
    "\n",
    "profiles {\n",
    "  conda_profile {\n",
    "    includeConfig 'conf/conda.config'\n",
    "    conda.enabled = true\n",
    "    executor = 'local'\n",
    "    }\n",
    "\n",
    "  docker_local {\n",
    "    includeConfig 'conf/docker.config'\n",
    "    docker.enabled = true\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    executor = 'local'\n",
    "  }\n",
    "  google_batch_profile {\n",
    "    workDir = 'gs://eila-nextflow-bucket/work/ATAC'\n",
    "    includeConfig 'conf/docker-google-batch.config'\n",
    "    docker.enabled = true\n",
    "    wave.enabled = true\n",
    "    fusion.enabled = true\n",
    "    process.scratch = false\n",
    "    executor = 'google-batch'\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    google.project = 'hai-gcp-genomic'\n",
    "    google.location = 'us-central1'\n",
    "    google.batch.spot = true\n",
    "    google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "  }\n",
    "    \n",
    "}\n",
    "\n",
    "// Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "\n",
    "params.FASTQS_SPEC_CH = '/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/atac_fastqs_files_vm.csv'\n",
    "params.CHROMAP_IDX='/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index'\n",
    "\n",
    "params.GENOME_FASTA='http://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz'\n",
    "params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "params.ENV_SYNAPSE_TOKEN = \"nextflow secrets get SYNAPSE\".execute().text.trim()\n",
    "\n",
    "params.CHROMAP_TRIM_ADAPTERS = false\n",
    "params.CHROMAP_REMOVE_PCR_DUPLICATES = false\n",
    "params.CHROMAP_REMOVE_PCR_DUPLICATES_AT_CELL_LEVEL = false\n",
    "params.CHROMAP_TN5_SHIFT = false\n",
    "params.CHROMAP_LOW_MEM = false\n",
    "params.CHROMAP_BED = false\n",
    "params.CHROMAP_MAX_INSERT_SIZE = 20\n",
    "params.CHROMAP_BC_ERROR_THRESHOLD = 20\n",
    "params.CHROMAP_BC_PROBABILITY_THRESHOLD = 0.20\n",
    "params.READ_FORMAT = 'r1:0:-1,bc:0:-1'\n",
    "params.DROP_REPETITIVE_THRESHOLD = false\n",
    "params.MAPQ_THRESHOLD = false\n",
    "params.BARCODE_TRANSLATE = 'path'\n",
    "//barcode_conversion_dict_file\n",
    "params.BARCODE_CONVERSION_DICT_FILE = '/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/na.na'\n",
    "params.SUBPOOL='  '\n",
    "params.ATAC_BARCODE_AND_POOL='atac_barcode_and_subpool.sh'\n",
    "params.ATAC_FRAGMENTS_CUTOFF = 4\n",
    "params.RUN_TABIX_SCRIPT='run_tabix.sh'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/na.na\n"
     ]
    }
   ],
   "source": [
    "!ls /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/na.na"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# // Try to trim adapters on 3 prime\n",
    "# params.CHROMAP_TRIM_ADAPTERS = false\n",
    "# // Remove PCR duplicates\n",
    "# params.CHROMAP_REMOVE_PCR_DUPLICATES = false\n",
    "# // Remove PCR duplicates at cell level for single cell data\n",
    "# params.CHROMAP_REMOVE_PCR_DUPLICATES_AT_CELL_LEVEL = false\n",
    "# // Perform Tn5 shift\n",
    "# params.CHROMAP_TN5_SHIFT = false\n",
    "# // Use low memory mode\n",
    "# params.CHROMAP_LOW_MEM = false\n",
    "# // Output mappings in BED/BEDPE format\n",
    "# params.CHROMAP_BED = false\n",
    "# //INT     Max insert size, only for paired-end read mapping [1000]\n",
    "# params.CHROMAP_MAX_INSERT_SIZE = 20\n",
    "# // INT  Max Hamming distance allowed to correct a barcode [1]\n",
    "# params.CHROMAP_BC_ERROR_THRESHOLD = 20\n",
    "# // FLT Min probability to correct a barcode [0.9]\n",
    "# params.CHROMAP_BC_PROBABILITY_THRESHOLD = 0.20\n",
    "# // STR Format for read files and barcode files [r1:0:-1,bc:0:-1 as 10x Genomics single-end format]\n",
    "# params.READ_FORMAT = 'r1:0:-1,bc:0:-1'\n",
    "# params.DROP_REPETITIVE_THRESHOLD = false\n",
    "# // INT  Min MAPQ in range [0, 60] for mappings to be output [30]\n",
    "# params.MAPQ_THRESHOLD = false\n",
    "# //FILE  Convert barcode to the specified sequences during output\n",
    "# params.BARCODE_TRANSLATE = 'path'\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# %%writefile nextflow_vm_anvil.config\n",
    "\n",
    "# includeConfig 'conf/conda.config'\n",
    "\n",
    "\n",
    "# manifest {\n",
    "#   description = 'A nexflow pipeline for fg single cell atac' \n",
    "#   mainScript = 'nf_wf_fg_singlecell_atac.nf'\n",
    "# }\n",
    "\n",
    "# profiles {\n",
    "#   conda_profile {\n",
    "#     includeConfig 'conf/conda.config'\n",
    "#     conda.enabled = true\n",
    "#     executor = 'local'\n",
    "#     }\n",
    "\n",
    "#   conda_docker_profile {\n",
    "#     includeConfig 'conf/conda_docker.config'\n",
    "#     executor = 'local'\n",
    "#     }\n",
    "\n",
    "#   docker_profile {\n",
    "#     includeConfig 'conf/docker.config'\n",
    "#     docker.enabled = true\n",
    "#     docker.envWhitelist=['TOWER_ACCESS_TOKEN']\n",
    "#     executor = 'local'\n",
    "#   }\n",
    "#   google_batch_profile {\n",
    "#     workDir = 'gs://fc-secure-87394b1b-e8d0-4802-9671-70bc3c8b7525/nextflow'\n",
    "#     includeConfig 'conf/docker-google-batch.config'\n",
    "#     docker.enabled = true\n",
    "#     docker.envWhitelist=['ENV_SYNAPSE_TOKEN']\n",
    "#     executor = 'google-batch'\n",
    "#     google.project = 'terra-addd4c49'\n",
    "#     location = 'us-central1'\n",
    "#     batch.spot = false\n",
    "#     batch.serviceAccountEmail='pet-26869426946125b815f6b@terra-addd4c49.iam.gserviceaccount.com'\n",
    "#   }\n",
    "# }\n",
    "\n",
    "# params.FASTQS_SPEC_CH = '/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/atac_fastqs_files_vm.csv'\n",
    "# params.CHROMAP_IDX='/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index'\n",
    "\n",
    "# params.GENOME_FASTA='http://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz'\n",
    "# params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm -r /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !rm -r /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !gsutil rm -r gs://eila-nextflow-bucket/work/ATAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm .nextflow*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !ls nextflow.config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.10.0\n",
      "Launching `nf_wf_fg_singlecell_atac.nf` [sleepy_neumann] DSL2 - revision: 8bcefc3068\n",
      "Dec 11, 2023 3:16:10 AM com.google.auth.oauth2.DefaultCredentialsProvider warnAboutProblematicCredentials\n",
      "WARNING: Your application has authenticated using end user credentials from Google Cloud SDK. We recommend that most server applications use service accounts instead. If your application continues to use end user credentials from Cloud SDK, you might receive a \"quota exceeded\" or \"API not enabled\" error. For more information about service accounts, see https://cloud.google.com/docs/authentication/.\n",
      "/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/atac_fastqs_files_vm.csv\n",
      "null\n",
      "[-        ] process > run_seqspec_modify_atac      -\u001b[K\n",
      "[-        ] process > run_seqspec_print            -\u001b[K\n",
      "[-        ] process > run_chromap_test             -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx       -\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix                    -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "[-        ] process > run_tss                      -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "\u001b[47A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_seqspec_modify_atac      -\u001b[K\n",
      "[-        ] process > run_seqspec_print            -\u001b[K\n",
      "[-        ] process > run_chromap_test             -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx       -\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix                    -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "[-        ] process > run_tss                      -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "\u001b[48A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_seqspec_modify_atac      -\u001b[K\n",
      "[-        ] process > run_seqspec_print            -\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test             [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx       -\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix                    -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "[-        ] process > run_tss                      -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "\u001b[48A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_seqspec_modify_atac      -\u001b[K\n",
      "[-        ] process > run_seqspec_print            -\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx       -\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix                    -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "[-        ] process > run_tss                      -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "\u001b[51A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)  [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_seqspec_print            -\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)   [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix                    -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "[-        ] process > run_tss                      -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "\u001b[51A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_seqspec_print            -\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)   [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix                    -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "[-        ] process > run_tss                      -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_atac\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "RNA-737K-arc-v1.txt\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify atac\u001b[K\n",
      "\u001b[K\n",
      "\u001b[63A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_seqspec_print            -\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix                    -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "[-        ] process > run_tss                      -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_atac\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "RNA-737K-arc-v1.txt\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify atac\u001b[K\n",
      "\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "fastq1 is atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "fastq2 is atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "fastq3 is atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "spec_yaml is nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "chromap_trim_adaptor is false\u001b[K\n",
      "task.cpus is 1. should be 32. TODO: check what is the issue\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "\u001b[K\n",
      "\u001b[73A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix                    -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "[-        ] process > run_tss                      -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_atac\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "RNA-737K-arc-v1.txt\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify atac\u001b[K\n",
      "\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "fastq1 is atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "fastq2 is atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "fastq3 is atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "spec_yaml is nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "chromap_trim_adaptor is false\u001b[K\n",
      "task.cpus is 1. should be 32. TODO: check what is the issue\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "\u001b[K\n",
      "\u001b[73A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_tabix                    -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "[-        ] process > run_tss                      -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_atac\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "RNA-737K-arc-v1.txt\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify atac\u001b[K\n",
      "\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "fastq1 is atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "fastq2 is atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "fastq3 is atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "spec_yaml is nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "chromap_trim_adaptor is false\u001b[K\n",
      "task.cpus is 1. should be 32. TODO: check what is the issue\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "\u001b[K\n",
      "\u001b[73A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_tabix                    -\u001b[K\n",
      "[1e/4afc9b] process > run_merge_logs (1)           [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "[-        ] process > run_tss                      -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_atac\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "RNA-737K-arc-v1.txt\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify atac\u001b[K\n",
      "\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "fastq1 is atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "fastq2 is atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "fastq3 is atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "spec_yaml is nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "chromap_trim_adaptor is false\u001b[K\n",
      "task.cpus is 1. should be 32. TODO: check what is the issue\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "\u001b[K\n",
      "start run_bgzip_fragment_file\u001b[K\n",
      "fragment_file is: map_bed_file.bed\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 92 Dec 11 01:10 map_bed_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/05/dd7a46358d40ce9c553a3e1b93bb77/map_bed_file.bed\u001b[K\n",
      "total 5201\u001b[K\n",
      "-rw-r--r-- 1 root root 5324959 Dec 11 01:10 map_bed_file.bed.gz\u001b[K\n",
      "finished run_bgzip_fragment_file\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: 112d4f3487d7\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "\u001b[89A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_tabix                    -\u001b[K\n",
      "[1e/4afc9b] process > run_merge_logs (1)           [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "[-        ] process > run_tss                      -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_atac\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "RNA-737K-arc-v1.txt\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify atac\u001b[K\n",
      "\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "fastq1 is atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "fastq2 is atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "fastq3 is atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "spec_yaml is nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "chromap_trim_adaptor is false\u001b[K\n",
      "task.cpus is 1. should be 32. TODO: check what is the issue\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "\u001b[K\n",
      "start run_bgzip_fragment_file\u001b[K\n",
      "fragment_file is: map_bed_file.bed\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 92 Dec 11 01:10 map_bed_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/05/dd7a46358d40ce9c553a3e1b93bb77/map_bed_file.bed\u001b[K\n",
      "total 5201\u001b[K\n",
      "-rw-r--r-- 1 root root 5324959 Dec 11 01:10 map_bed_file.bed.gz\u001b[K\n",
      "finished run_bgzip_fragment_file\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: 112d4f3487d7\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "\u001b[113A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[87/dcf012] process > run_tabix (1)                [100%] 1 of 1, cached: 1\u001b[K\n",
      "[1e/4afc9b] process > run_merge_logs (1)           [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "[-        ] process > run_tss                      -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_atac\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "RNA-737K-arc-v1.txt\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify atac\u001b[K\n",
      "\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "fastq1 is atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "fastq2 is atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "fastq3 is atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "spec_yaml is nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "chromap_trim_adaptor is false\u001b[K\n",
      "task.cpus is 1. should be 32. TODO: check what is the issue\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "\u001b[K\n",
      "start run_bgzip_fragment_file\u001b[K\n",
      "fragment_file is: map_bed_file.bed\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 92 Dec 11 01:10 map_bed_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/05/dd7a46358d40ce9c553a3e1b93bb77/map_bed_file.bed\u001b[K\n",
      "total 5201\u001b[K\n",
      "-rw-r--r-- 1 root root 5324959 Dec 11 01:10 map_bed_file.bed.gz\u001b[K\n",
      "finished run_bgzip_fragment_file\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: 112d4f3487d7\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "\u001b[113A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[87/dcf012] process > run_tabix (1)                [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[1e/4afc9b] process > run_merge_logs (1)           [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "[-        ] process > run_tss                      -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_atac\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "RNA-737K-arc-v1.txt\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify atac\u001b[K\n",
      "\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "fastq1 is atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "fastq2 is atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "fastq3 is atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "spec_yaml is nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "chromap_trim_adaptor is false\u001b[K\n",
      "task.cpus is 1. should be 32. TODO: check what is the issue\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "\u001b[K\n",
      "start run_bgzip_fragment_file\u001b[K\n",
      "fragment_file is: map_bed_file.bed\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 92 Dec 11 01:10 map_bed_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/05/dd7a46358d40ce9c553a3e1b93bb77/map_bed_file.bed\u001b[K\n",
      "total 5201\u001b[K\n",
      "-rw-r--r-- 1 root root 5324959 Dec 11 01:10 map_bed_file.bed.gz\u001b[K\n",
      "finished run_bgzip_fragment_file\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: 112d4f3487d7\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "atac_barcode_and_subpool.sh\u001b[K\n",
      "create_sample_filtered_fragment_file.sh\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 95 Dec 11 01:15 map_bed_file.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/7b/9d8290454e4f2df93c255dc6a83c2c/map_bed_file.bed.gz\u001b[K\n",
      "total 662\u001b[K\n",
      "lrwxrwxrwx 1 root root     95 Dec 11 01:15 map_bed_file.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/7b/9d8290454e4f2df93c255dc6a83c2c/map_bed_file.bed.gz\u001b[K\n",
      "-rw-r--r-- 1 root root 677057 Dec 11 01:15 map_bed_file.bed.gz.tbi\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "\u001b[K\n",
      "\u001b[126A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[87/dcf012] process > run_tabix (1)                [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[1e/4afc9b] process > run_merge_logs (1)           [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[e1/008d5d] process > run_filter_fragments (1)     [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "[-        ] process > run_tss                      -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_atac\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "RNA-737K-arc-v1.txt\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify atac\u001b[K\n",
      "\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "fastq1 is atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "fastq2 is atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "fastq3 is atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "spec_yaml is nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "chromap_trim_adaptor is false\u001b[K\n",
      "task.cpus is 1. should be 32. TODO: check what is the issue\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "\u001b[K\n",
      "start run_bgzip_fragment_file\u001b[K\n",
      "fragment_file is: map_bed_file.bed\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 92 Dec 11 01:10 map_bed_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/05/dd7a46358d40ce9c553a3e1b93bb77/map_bed_file.bed\u001b[K\n",
      "total 5201\u001b[K\n",
      "-rw-r--r-- 1 root root 5324959 Dec 11 01:10 map_bed_file.bed.gz\u001b[K\n",
      "finished run_bgzip_fragment_file\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: 112d4f3487d7\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "atac_barcode_and_subpool.sh\u001b[K\n",
      "create_sample_filtered_fragment_file.sh\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 95 Dec 11 01:15 map_bed_file.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/7b/9d8290454e4f2df93c255dc6a83c2c/map_bed_file.bed.gz\u001b[K\n",
      "total 662\u001b[K\n",
      "lrwxrwxrwx 1 root root     95 Dec 11 01:15 map_bed_file.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/7b/9d8290454e4f2df93c255dc6a83c2c/map_bed_file.bed.gz\u001b[K\n",
      "-rw-r--r-- 1 root root 677057 Dec 11 01:15 map_bed_file.bed.gz.tbi\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "\u001b[K\n",
      "\u001b[126A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)   [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[87/dcf012] process > run_tabix (1)                [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[1e/4afc9b] process > run_merge_logs (1)           [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[e1/008d5d] process > run_filter_fragments (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "[-        ] process > run_tss                      -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_atac\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "RNA-737K-arc-v1.txt\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify atac\u001b[K\n",
      "\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "fastq1 is atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "fastq2 is atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "fastq3 is atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "spec_yaml is nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "chromap_trim_adaptor is false\u001b[K\n",
      "task.cpus is 1. should be 32. TODO: check what is the issue\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "\u001b[K\n",
      "start run_bgzip_fragment_file\u001b[K\n",
      "fragment_file is: map_bed_file.bed\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 92 Dec 11 01:10 map_bed_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/05/dd7a46358d40ce9c553a3e1b93bb77/map_bed_file.bed\u001b[K\n",
      "total 5201\u001b[K\n",
      "-rw-r--r-- 1 root root 5324959 Dec 11 01:10 map_bed_file.bed.gz\u001b[K\n",
      "finished run_bgzip_fragment_file\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: 112d4f3487d7\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "atac_barcode_and_subpool.sh\u001b[K\n",
      "create_sample_filtered_fragment_file.sh\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 95 Dec 11 01:15 map_bed_file.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/7b/9d8290454e4f2df93c255dc6a83c2c/map_bed_file.bed.gz\u001b[K\n",
      "total 662\u001b[K\n",
      "lrwxrwxrwx 1 root root     95 Dec 11 01:15 map_bed_file.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/7b/9d8290454e4f2df93c255dc6a83c2c/map_bed_file.bed.gz\u001b[K\n",
      "-rw-r--r-- 1 root root 677057 Dec 11 01:15 map_bed_file.bed.gz.tbi\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_fragments\u001b[K\n",
      "barcode_conversion_dict_file is na.na\u001b[K\n",
      "atac_and_pool_bash_script is atac_barcode_and_subpool.sh\u001b[K\n",
      "call to ls usr local bin\u001b[K\n",
      "atac_barcode_and_subpool.sh\u001b[K\n",
      "create_sample_filtered_fragment_file.sh\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "------ Script Parameters ------\u001b[K\n",
      "barcode_conversion_dict_file: na.na\u001b[K\n",
      "subpool_in: summary_file.csv\u001b[K\n",
      "barcode_summary: 4\u001b[K\n",
      "fragments_cutoff: map_bed_file.bed.gz.tbi\u001b[K\n",
      "fragments_file: 1\u001b[K\n",
      "cpus: filtered_fragment_file.bed\u001b[K\n",
      "filtered_fragment_file: \u001b[K\n",
      "------------------------------\u001b[K\n",
      "------ There is no barcode_conversion_dict_file ------\u001b[K\n",
      "------ Number of barcodes BEFORE filtering ------\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "0\u001b[K\n",
      "TODO:comment out when execute with real data\u001b[K\n",
      "chr1\t100\t200\tfragment1\t1\t+\u001b[K\n",
      "chr2\t300\t400\tfragment2\t2\t-\u001b[K\n",
      "chr3\t500\t600\tfragment3\t3\t+\u001b[K\n",
      "ls after the execution of the script\u001b[K\n",
      "filtered_fragment_file.bed\u001b[K\n",
      "map_bed_file.bed.gz.tbi\u001b[K\n",
      "na.na\u001b[K\n",
      "summary_file.csv\u001b[K\n",
      "end run_filter_fragments\u001b[K\n",
      "\u001b[K\n",
      "\u001b[158A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)    [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[87/dcf012] process > run_tabix (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[1e/4afc9b] process > run_merge_logs (1)             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[e1/008d5d] process > run_filter_fragments (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[c1/1f5aad] process > run_tabix_filtered_fragment... [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_tss                        -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_atac\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "RNA-737K-arc-v1.txt\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify atac\u001b[K\n",
      "\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "fastq1 is atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "fastq2 is atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "fastq3 is atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "spec_yaml is nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "chromap_trim_adaptor is false\u001b[K\n",
      "task.cpus is 1. should be 32. TODO: check what is the issue\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "\u001b[K\n",
      "start run_bgzip_fragment_file\u001b[K\n",
      "fragment_file is: map_bed_file.bed\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 92 Dec 11 01:10 map_bed_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/05/dd7a46358d40ce9c553a3e1b93bb77/map_bed_file.bed\u001b[K\n",
      "total 5201\u001b[K\n",
      "-rw-r--r-- 1 root root 5324959 Dec 11 01:10 map_bed_file.bed.gz\u001b[K\n",
      "finished run_bgzip_fragment_file\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: 112d4f3487d7\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "atac_barcode_and_subpool.sh\u001b[K\n",
      "create_sample_filtered_fragment_file.sh\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 95 Dec 11 01:15 map_bed_file.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/7b/9d8290454e4f2df93c255dc6a83c2c/map_bed_file.bed.gz\u001b[K\n",
      "total 662\u001b[K\n",
      "lrwxrwxrwx 1 root root     95 Dec 11 01:15 map_bed_file.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/7b/9d8290454e4f2df93c255dc6a83c2c/map_bed_file.bed.gz\u001b[K\n",
      "-rw-r--r-- 1 root root 677057 Dec 11 01:15 map_bed_file.bed.gz.tbi\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_fragments\u001b[K\n",
      "barcode_conversion_dict_file is na.na\u001b[K\n",
      "atac_and_pool_bash_script is atac_barcode_and_subpool.sh\u001b[K\n",
      "call to ls usr local bin\u001b[K\n",
      "atac_barcode_and_subpool.sh\u001b[K\n",
      "create_sample_filtered_fragment_file.sh\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "------ Script Parameters ------\u001b[K\n",
      "barcode_conversion_dict_file: na.na\u001b[K\n",
      "subpool_in: summary_file.csv\u001b[K\n",
      "barcode_summary: 4\u001b[K\n",
      "fragments_cutoff: map_bed_file.bed.gz.tbi\u001b[K\n",
      "fragments_file: 1\u001b[K\n",
      "cpus: filtered_fragment_file.bed\u001b[K\n",
      "filtered_fragment_file: \u001b[K\n",
      "------------------------------\u001b[K\n",
      "------ There is no barcode_conversion_dict_file ------\u001b[K\n",
      "------ Number of barcodes BEFORE filtering ------\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "0\u001b[K\n",
      "TODO:comment out when execute with real data\u001b[K\n",
      "chr1\t100\t200\tfragment1\t1\t+\u001b[K\n",
      "chr2\t300\t400\tfragment2\t2\t-\u001b[K\n",
      "chr3\t500\t600\tfragment3\t3\t+\u001b[K\n",
      "ls after the execution of the script\u001b[K\n",
      "filtered_fragment_file.bed\u001b[K\n",
      "map_bed_file.bed.gz.tbi\u001b[K\n",
      "na.na\u001b[K\n",
      "summary_file.csv\u001b[K\n",
      "end run_filter_fragments\u001b[K\n",
      "\u001b[K\n",
      "\u001b[158A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)    [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[87/dcf012] process > run_tabix (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[1e/4afc9b] process > run_merge_logs (1)             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[e1/008d5d] process > run_filter_fragments (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[c1/1f5aad] process > run_tabix_filtered_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_tss                        -\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_atac\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "RNA-737K-arc-v1.txt\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify atac\u001b[K\n",
      "\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "fastq1 is atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "fastq2 is atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "fastq3 is atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "spec_yaml is nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "chromap_trim_adaptor is false\u001b[K\n",
      "task.cpus is 1. should be 32. TODO: check what is the issue\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "\u001b[K\n",
      "start run_bgzip_fragment_file\u001b[K\n",
      "fragment_file is: map_bed_file.bed\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 92 Dec 11 01:10 map_bed_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/05/dd7a46358d40ce9c553a3e1b93bb77/map_bed_file.bed\u001b[K\n",
      "total 5201\u001b[K\n",
      "-rw-r--r-- 1 root root 5324959 Dec 11 01:10 map_bed_file.bed.gz\u001b[K\n",
      "finished run_bgzip_fragment_file\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: 112d4f3487d7\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "atac_barcode_and_subpool.sh\u001b[K\n",
      "create_sample_filtered_fragment_file.sh\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 95 Dec 11 01:15 map_bed_file.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/7b/9d8290454e4f2df93c255dc6a83c2c/map_bed_file.bed.gz\u001b[K\n",
      "total 662\u001b[K\n",
      "lrwxrwxrwx 1 root root     95 Dec 11 01:15 map_bed_file.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/7b/9d8290454e4f2df93c255dc6a83c2c/map_bed_file.bed.gz\u001b[K\n",
      "-rw-r--r-- 1 root root 677057 Dec 11 01:15 map_bed_file.bed.gz.tbi\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_fragments\u001b[K\n",
      "barcode_conversion_dict_file is na.na\u001b[K\n",
      "atac_and_pool_bash_script is atac_barcode_and_subpool.sh\u001b[K\n",
      "call to ls usr local bin\u001b[K\n",
      "atac_barcode_and_subpool.sh\u001b[K\n",
      "create_sample_filtered_fragment_file.sh\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "------ Script Parameters ------\u001b[K\n",
      "barcode_conversion_dict_file: na.na\u001b[K\n",
      "subpool_in: summary_file.csv\u001b[K\n",
      "barcode_summary: 4\u001b[K\n",
      "fragments_cutoff: map_bed_file.bed.gz.tbi\u001b[K\n",
      "fragments_file: 1\u001b[K\n",
      "cpus: filtered_fragment_file.bed\u001b[K\n",
      "filtered_fragment_file: \u001b[K\n",
      "------------------------------\u001b[K\n",
      "------ There is no barcode_conversion_dict_file ------\u001b[K\n",
      "------ Number of barcodes BEFORE filtering ------\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "0\u001b[K\n",
      "TODO:comment out when execute with real data\u001b[K\n",
      "chr1\t100\t200\tfragment1\t1\t+\u001b[K\n",
      "chr2\t300\t400\tfragment2\t2\t-\u001b[K\n",
      "chr3\t500\t600\tfragment3\t3\t+\u001b[K\n",
      "ls after the execution of the script\u001b[K\n",
      "filtered_fragment_file.bed\u001b[K\n",
      "map_bed_file.bed.gz.tbi\u001b[K\n",
      "na.na\u001b[K\n",
      "summary_file.csv\u001b[K\n",
      "end run_filter_fragments\u001b[K\n",
      "\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "atac_barcode_and_subpool.sh\u001b[K\n",
      "create_sample_filtered_fragment_file.sh\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 102 Dec 11 03:08 filtered_fragment_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/e1/008d5dcd22994dbab9b505cabba8d9/filtered_fragment_file.bed\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 102 Dec 11 03:08 filtered_fragment_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/e1/008d5dcd22994dbab9b505cabba8d9/filtered_fragment_file.bed\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "create the output file with the nextflow output variable\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "\u001b[K\n",
      "\u001b[171A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)    [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[87/dcf012] process > run_tabix (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[1e/4afc9b] process > run_merge_logs (1)             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[e1/008d5d] process > run_filter_fragments (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[c1/1f5aad] process > run_tabix_filtered_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_tss                        [  0%] 0 of 1\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_atac\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "RNA-737K-arc-v1.txt\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify atac\u001b[K\n",
      "\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "fastq1 is atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "fastq2 is atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "fastq3 is atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "spec_yaml is nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "chromap_trim_adaptor is false\u001b[K\n",
      "task.cpus is 1. should be 32. TODO: check what is the issue\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "\u001b[K\n",
      "start run_bgzip_fragment_file\u001b[K\n",
      "fragment_file is: map_bed_file.bed\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 92 Dec 11 01:10 map_bed_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/05/dd7a46358d40ce9c553a3e1b93bb77/map_bed_file.bed\u001b[K\n",
      "total 5201\u001b[K\n",
      "-rw-r--r-- 1 root root 5324959 Dec 11 01:10 map_bed_file.bed.gz\u001b[K\n",
      "finished run_bgzip_fragment_file\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: 112d4f3487d7\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "atac_barcode_and_subpool.sh\u001b[K\n",
      "create_sample_filtered_fragment_file.sh\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 95 Dec 11 01:15 map_bed_file.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/7b/9d8290454e4f2df93c255dc6a83c2c/map_bed_file.bed.gz\u001b[K\n",
      "total 662\u001b[K\n",
      "lrwxrwxrwx 1 root root     95 Dec 11 01:15 map_bed_file.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/7b/9d8290454e4f2df93c255dc6a83c2c/map_bed_file.bed.gz\u001b[K\n",
      "-rw-r--r-- 1 root root 677057 Dec 11 01:15 map_bed_file.bed.gz.tbi\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_fragments\u001b[K\n",
      "barcode_conversion_dict_file is na.na\u001b[K\n",
      "atac_and_pool_bash_script is atac_barcode_and_subpool.sh\u001b[K\n",
      "call to ls usr local bin\u001b[K\n",
      "atac_barcode_and_subpool.sh\u001b[K\n",
      "create_sample_filtered_fragment_file.sh\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "------ Script Parameters ------\u001b[K\n",
      "barcode_conversion_dict_file: na.na\u001b[K\n",
      "subpool_in: summary_file.csv\u001b[K\n",
      "barcode_summary: 4\u001b[K\n",
      "fragments_cutoff: map_bed_file.bed.gz.tbi\u001b[K\n",
      "fragments_file: 1\u001b[K\n",
      "cpus: filtered_fragment_file.bed\u001b[K\n",
      "filtered_fragment_file: \u001b[K\n",
      "------------------------------\u001b[K\n",
      "------ There is no barcode_conversion_dict_file ------\u001b[K\n",
      "------ Number of barcodes BEFORE filtering ------\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "0\u001b[K\n",
      "TODO:comment out when execute with real data\u001b[K\n",
      "chr1\t100\t200\tfragment1\t1\t+\u001b[K\n",
      "chr2\t300\t400\tfragment2\t2\t-\u001b[K\n",
      "chr3\t500\t600\tfragment3\t3\t+\u001b[K\n",
      "ls after the execution of the script\u001b[K\n",
      "filtered_fragment_file.bed\u001b[K\n",
      "map_bed_file.bed.gz.tbi\u001b[K\n",
      "na.na\u001b[K\n",
      "summary_file.csv\u001b[K\n",
      "end run_filter_fragments\u001b[K\n",
      "\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "atac_barcode_and_subpool.sh\u001b[K\n",
      "create_sample_filtered_fragment_file.sh\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 102 Dec 11 03:08 filtered_fragment_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/e1/008d5dcd22994dbab9b505cabba8d9/filtered_fragment_file.bed\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 102 Dec 11 03:08 filtered_fragment_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/e1/008d5dcd22994dbab9b505cabba8d9/filtered_fragment_file.bed\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "create the output file with the nextflow output variable\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "\u001b[K\n",
      "\u001b[171A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)    [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[87/dcf012] process > run_tabix (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[1e/4afc9b] process > run_merge_logs (1)             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[e1/008d5d] process > run_filter_fragments (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[c1/1f5aad] process > run_tabix_filtered_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[0d/b92d4b] process > run_tss (1)                    [  0%] 0 of 1\u001b[K\n",
      "after run_seqspec_modify_atac\u001b[K\n",
      "after run_seqspec_print\u001b[K\n",
      "after run_chromap_test\u001b[K\n",
      "after genome_fasta_ch\u001b[K\n",
      "after genome_gtf_ch\u001b[K\n",
      "after genome_chromap_idx download\u001b[K\n",
      "chromap parameters\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "20\u001b[K\n",
      "20\u001b[K\n",
      "0.20\u001b[K\n",
      "r1:0:-1,bc:0:-1\u001b[K\n",
      "false\u001b[K\n",
      "false\u001b[K\n",
      "path\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "before call run_tabix_fragment_file\u001b[K\n",
      "after call run_tabix_fragment_file\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "DataflowBroadcast around DataflowStream[?]\u001b[K\n",
      "before call run_tss\u001b[K\n",
      "after call run_tss\u001b[K\n",
      "run_chromap_test\u001b[K\n",
      "run_chromap_test_end\u001b[K\n",
      "\u001b[K\n",
      "start run_seqspec_modify_atac\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "RNA-737K-arc-v1.txt\u001b[K\n",
      "atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "finished seqspec modify atac\u001b[K\n",
      "\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index\u001b[K\n",
      "Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\u001b[K\n",
      "fastq1 is atac_R1_SRR18677642.fastq.gz\u001b[K\n",
      "fastq2 is atac_R2_SRR18677642.fastq.gz\u001b[K\n",
      "fastq3 is atac_R3_SRR18677642.fastq.gz\u001b[K\n",
      "spec_yaml is nf_spec_dogmaseq_dig.yaml\u001b[K\n",
      "chromap_trim_adaptor is false\u001b[K\n",
      "task.cpus is 1. should be 32. TODO: check what is the issue\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "\u001b[K\n",
      "start run_bgzip_fragment_file\u001b[K\n",
      "fragment_file is: map_bed_file.bed\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 92 Dec 11 01:10 map_bed_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/05/dd7a46358d40ce9c553a3e1b93bb77/map_bed_file.bed\u001b[K\n",
      "total 5201\u001b[K\n",
      "-rw-r--r-- 1 root root 5324959 Dec 11 01:10 map_bed_file.bed.gz\u001b[K\n",
      "finished run_bgzip_fragment_file\u001b[K\n",
      "\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "I am running inside container: 112d4f3487d7\u001b[K\n",
      "start run_seqspec_print\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "nf_seqspec_modify.yaml\u001b[K\n",
      "seqspec.print.out\u001b[K\n",
      "finished seqspec print\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,2000000\u001b[K\n",
      "Number_of_mapped_reads,1849960\u001b[K\n",
      "Number_of_uniquely_mapped_reads,1765184\u001b[K\n",
      "Number_of_reads_have_multi-mappings,84776\u001b[K\n",
      "Number_of_candidates,7559730\u001b[K\n",
      "Number_of_mappings,1849960\u001b[K\n",
      "Number_of_uni-mappings,1765184\u001b[K\n",
      "Number_of_multi-mappings,84776\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),861791\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "atac_barcode_and_subpool.sh\u001b[K\n",
      "create_sample_filtered_fragment_file.sh\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 95 Dec 11 01:15 map_bed_file.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/7b/9d8290454e4f2df93c255dc6a83c2c/map_bed_file.bed.gz\u001b[K\n",
      "total 662\u001b[K\n",
      "lrwxrwxrwx 1 root root     95 Dec 11 01:15 map_bed_file.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/7b/9d8290454e4f2df93c255dc6a83c2c/map_bed_file.bed.gz\u001b[K\n",
      "-rw-r--r-- 1 root root 677057 Dec 11 01:15 map_bed_file.bed.gz.tbi\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_fragments\u001b[K\n",
      "barcode_conversion_dict_file is na.na\u001b[K\n",
      "atac_and_pool_bash_script is atac_barcode_and_subpool.sh\u001b[K\n",
      "call to ls usr local bin\u001b[K\n",
      "atac_barcode_and_subpool.sh\u001b[K\n",
      "create_sample_filtered_fragment_file.sh\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "------ Script Parameters ------\u001b[K\n",
      "barcode_conversion_dict_file: na.na\u001b[K\n",
      "subpool_in: summary_file.csv\u001b[K\n",
      "barcode_summary: 4\u001b[K\n",
      "fragments_cutoff: map_bed_file.bed.gz.tbi\u001b[K\n",
      "fragments_file: 1\u001b[K\n",
      "cpus: filtered_fragment_file.bed\u001b[K\n",
      "filtered_fragment_file: \u001b[K\n",
      "------------------------------\u001b[K\n",
      "------ There is no barcode_conversion_dict_file ------\u001b[K\n",
      "------ Number of barcodes BEFORE filtering ------\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "0\u001b[K\n",
      "TODO:comment out when execute with real data\u001b[K\n",
      "chr1\t100\t200\tfragment1\t1\t+\u001b[K\n",
      "chr2\t300\t400\tfragment2\t2\t-\u001b[K\n",
      "chr3\t500\t600\tfragment3\t3\t+\u001b[K\n",
      "ls after the execution of the script\u001b[K\n",
      "filtered_fragment_file.bed\u001b[K\n",
      "map_bed_file.bed.gz.tbi\u001b[K\n",
      "na.na\u001b[K\n",
      "summary_file.csv\u001b[K\n",
      "end run_filter_fragments\u001b[K\n",
      "\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "atac_barcode_and_subpool.sh\u001b[K\n",
      "create_sample_filtered_fragment_file.sh\u001b[K\n",
      "run_tabix.sh\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 102 Dec 11 03:08 filtered_fragment_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/e1/008d5dcd22994dbab9b505cabba8d9/filtered_fragment_file.bed\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 102 Dec 11 03:08 filtered_fragment_file.bed -> /fusion/gs/eila-nextflow-bucket/work/ATAC/e1/008d5dcd22994dbab9b505cabba8d9/filtered_fragment_file.bed\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "create the output file with the nextflow output variable\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "\u001b[K\n",
      "\u001b[172A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)    [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[87/dcf012] process > run_tabix (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[1e/4afc9b] process > run_merge_logs (1)             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[e1/008d5d] process > run_filter_fragments (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[c1/1f5aad] process > run_tabix_filtered_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[0d/b92d4b] process > run_tss (1)                    [  0%] 0 of 1\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[172A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)    [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[87/dcf012] process > run_tabix (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[1e/4afc9b] process > run_merge_logs (1)             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[e1/008d5d] process > run_filter_fragments (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[c1/1f5aad] process > run_tabix_filtered_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[0d/b92d4b] process > run_tss (1)                    [  0%] 0 of 1\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'run_tss (1)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Missing output file(s) `filtered_fragment_file.bed.tbi.tss` expected by process `run_tss (1)`\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  echo 'start run_tss'\u001b[K\n",
      "  ls /usr/local/bin/\u001b[K\n",
      "  \u001b[K\n",
      "  echo 'finished run_tss'\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  0\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  start run_tss\u001b[K\n",
      "  atac_barcode_and_subpool.sh\u001b[K\n",
      "  create_sample_filtered_fragment_file.sh\u001b[K\n",
      "  run_tabix.sh\u001b[K\n",
      "  finished run_tss\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  gs://eila-nextflow-bucket/work/ATAC/0d/b92d4b523d1aab1a4b799fb3b8a7fd\u001b[K\n",
      "\u001b[K\n",
      "Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[172A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)    [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[87/dcf012] process > run_tabix (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[1e/4afc9b] process > run_merge_logs (1)             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[e1/008d5d] process > run_filter_fragments (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[c1/1f5aad] process > run_tabix_filtered_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[0d/b92d4b] process > run_tss (1)                    [  0%] 0 of 1 ✔\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'run_tss (1)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Missing output file(s) `filtered_fragment_file.bed.tbi.tss` expected by process `run_tss (1)`\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  echo 'start run_tss'\u001b[K\n",
      "  ls /usr/local/bin/\u001b[K\n",
      "  \u001b[K\n",
      "  echo 'finished run_tss'\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  0\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  start run_tss\u001b[K\n",
      "  atac_barcode_and_subpool.sh\u001b[K\n",
      "  create_sample_filtered_fragment_file.sh\u001b[K\n",
      "  run_tabix.sh\u001b[K\n",
      "  finished run_tss\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  gs://eila-nextflow-bucket/work/ATAC/0d/b92d4b523d1aab1a4b799fb3b8a7fd\u001b[K\n",
      "\u001b[K\n",
      "Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)    [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[87/dcf012] process > run_tabix (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[1e/4afc9b] process > run_merge_logs (1)             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[e1/008d5d] process > run_filter_fragments (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[c1/1f5aad] process > run_tabix_filtered_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[0d/b92d4b] process > run_tss (1)                    [100%] 1 of 1, failed: 1 ✘\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'run_tss (1)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Missing output file(s) `filtered_fragment_file.bed.tbi.tss` expected by process `run_tss (1)`\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  echo 'start run_tss'\u001b[K\n",
      "  ls /usr/local/bin/\u001b[K\n",
      "  \u001b[K\n",
      "  echo 'finished run_tss'\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  0\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  start run_tss\u001b[K\n",
      "  atac_barcode_and_subpool.sh\u001b[K\n",
      "  create_sample_filtered_fragment_file.sh\u001b[K\n",
      "  run_tabix.sh\u001b[K\n",
      "  finished run_tss\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  gs://eila-nextflow-bucket/work/ATAC/0d/b92d4b523d1aab1a4b799fb3b8a7fd\u001b[K\n",
      "\u001b[K\n",
      "Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[c8/7e4656] process > run_seqspec_modify_atac (1)    [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[25/135075] process > run_seqspec_print (1)          [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[db/6ecefc] process > run_chromap_test               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[05/dd7a46] process > run_chromap_map_to_idx (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/9d8290] process > run_bgzip (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[87/dcf012] process > run_tabix (1)                  [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[1e/4afc9b] process > run_merge_logs (1)             [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[e1/008d5d] process > run_filter_fragments (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[c1/1f5aad] process > run_tabix_filtered_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[0d/b92d4b] process > run_tss (1)                    [100%] 1 of 1, failed: 1 ✘\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'run_tss (1)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Missing output file(s) `filtered_fragment_file.bed.tbi.tss` expected by process `run_tss (1)`\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  echo 'start run_tss'\u001b[K\n",
      "  ls /usr/local/bin/\u001b[K\n",
      "  \u001b[K\n",
      "  echo 'finished run_tss'\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  0\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  start run_tss\u001b[K\n",
      "  atac_barcode_and_subpool.sh\u001b[K\n",
      "  create_sample_filtered_fragment_file.sh\u001b[K\n",
      "  run_tabix.sh\u001b[K\n",
      "  finished run_tss\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  gs://eila-nextflow-bucket/work/ATAC/0d/b92d4b523d1aab1a4b799fb3b8a7fd\u001b[K\n",
      "\u001b[K\n",
      "Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile -with-tower -resume\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm_anvil.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile docker_profile\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dec-11 03:16:06.003 [main] DEBUG nextflow.cli.Launcher - $> nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile -with-tower -resume\n",
      "Dec-11 03:16:06.327 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.10.0\n",
      "Dec-11 03:16:06.358 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/home/eila/.nextflow/plugins; core-plugins: nf-amazon@2.1.4,nf-azure@1.3.2,nf-cloudcache@0.3.0,nf-codecommit@0.1.5,nf-console@1.0.6,nf-ga4gh@1.1.0,nf-google@1.8.3,nf-tower@1.6.3,nf-wave@1.0.0\n",
      "Dec-11 03:16:06.375 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Enabled plugins: []\n",
      "Dec-11 03:16:06.377 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Disabled plugins: []\n",
      "Dec-11 03:16:06.383 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode\n",
      "Dec-11 03:16:06.400 [main] INFO  org.pf4j.AbstractPluginManager - No plugins\n",
      "Dec-11 03:16:06.460 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_vm.config\n",
      "Dec-11 03:16:06.461 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_vm.config\n",
      "Dec-11 03:16:06.503 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Dec-11 03:16:09.539 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [docker_local, google_batch_profile, conda_profile]\n",
      "Dec-11 03:16:09.595 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 by global default\n",
      "Dec-11 03:16:09.629 [main] INFO  nextflow.cli.CmdRun - Launching `nf_wf_fg_singlecell_atac.nf` [sleepy_neumann] DSL2 - revision: 8bcefc3068\n",
      "Dec-11 03:16:09.630 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[nf-google@1.8.3]\n",
      "Dec-11 03:16:09.632 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[nf-google@1.8.3, nf-tower@1.6.3, nf-wave@1.0.0]\n",
      "Dec-11 03:16:09.633 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-google version: 1.8.3\n",
      "Dec-11 03:16:09.657 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-google@1.8.3' resolved\n",
      "Dec-11 03:16:09.658 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-google@1.8.3'\n",
      "Dec-11 03:16:09.728 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-google@1.8.3\n",
      "Dec-11 03:16:09.728 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-tower version: 1.6.3\n",
      "Dec-11 03:16:09.732 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-tower@1.6.3' resolved\n",
      "Dec-11 03:16:09.732 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-tower@1.6.3'\n",
      "Dec-11 03:16:09.749 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-tower@1.6.3\n",
      "Dec-11 03:16:09.749 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-wave version: 1.0.0\n",
      "Dec-11 03:16:09.753 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-wave@1.0.0' resolved\n",
      "Dec-11 03:16:09.754 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-wave@1.0.0'\n",
      "Dec-11 03:16:09.767 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-wave@1.0.0\n",
      "Dec-11 03:16:09.795 [main] DEBUG n.secret.LocalSecretsProvider - Secrets store: /home/eila/.nextflow/secrets/store.json\n",
      "Dec-11 03:16:09.802 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@5d3b58ca] - activable => nextflow.secret.LocalSecretsProvider@5d3b58ca\n",
      "Dec-11 03:16:09.926 [main] DEBUG nextflow.Session - Session UUID: f7e37c4b-c1ab-4ccb-a8cb-153d209e8602\n",
      "Dec-11 03:16:09.926 [main] DEBUG nextflow.Session - Run name: sleepy_neumann\n",
      "Dec-11 03:16:09.927 [main] DEBUG nextflow.Session - Executor pool size: 4\n",
      "Dec-11 03:16:10.364 [main] DEBUG nextflow.file.FilePorter - File porter settings maxRetries=3; maxTransfers=50; pollTimeout=null\n",
      "Dec-11 03:16:10.375 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=12; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false\n",
      "Dec-11 03:16:10.383 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_vm.config\n",
      "Dec-11 03:16:10.383 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_vm.config\n",
      "Dec-11 03:16:10.384 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Dec-11 03:16:12.384 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [docker_local, google_batch_profile, conda_profile]\n",
      "Dec-11 03:16:12.431 [main] DEBUG nextflow.cli.CmdRun - \n",
      "  Version: 23.10.0 build 5889\n",
      "  Created: 15-10-2023 15:07 UTC \n",
      "  System: Linux 5.10.0-26-cloud-amd64\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 11.0.13+7-b1751.21\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "  Process: 141231@eila-vm [10.128.0.63]\n",
      "  CPUs: 4 - Mem: 14.6 GB (567.8 MB) - Swap: 0 (0)\n",
      "Dec-11 03:16:12.532 [main] DEBUG nextflow.file.FileHelper - Can't check if specified path is NFS (1): gs://eila-nextflow-bucket/work/ATAC\n",
      "\n",
      "Dec-11 03:16:12.533 [main] DEBUG nextflow.Session - Work-dir: gs://eila-nextflow-bucket/work/ATAC [null]\n",
      "Dec-11 03:16:12.571 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[GoogleLifeSciencesExecutor, GoogleBatchExecutor]\n",
      "Dec-11 03:16:12.592 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory\n",
      "Dec-11 03:16:12.595 [main] DEBUG nextflow.Session - Observer factory: WaveFactory\n",
      "Dec-11 03:16:12.596 [main] DEBUG io.seqera.wave.plugin.WaveFactory - Detected Fusion enabled -- Enabling bundle project resources -- Disabling upload of remote bin directory\n",
      "Dec-11 03:16:12.623 [main] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Dec-11 03:16:12.635 [main] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null, dryRunMode:false)\n",
      "Dec-11 03:16:12.783 [main] DEBUG nextflow.Session - Observer factory: TowerFactory\n",
      "Dec-11 03:16:12.872 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory\n",
      "Dec-11 03:16:12.887 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 5; maxThreads: 1000\n",
      "Dec-11 03:16:13.018 [main] DEBUG nextflow.Session - Session start\n",
      "Dec-11 03:16:13.033 [main] DEBUG io.seqera.tower.plugin.TowerClient - Creating Tower observer -- endpoint=https://api.tower.nf; requestInterval=1s; aliveInterval=1m; maxRetries=5; backOffBase=3; backOffDelay=250\n",
      "Dec-11 03:16:14.320 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution\n",
      "Dec-11 03:16:15.195 [main] INFO  nextflow.script.BaseScript - /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/atac_fastqs_files_vm.csv\n",
      "Dec-11 03:16:15.196 [main] INFO  nextflow.script.BaseScript - null\n",
      "Dec-11 03:16:15.437 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:seqspec_local` matches labels `seqspec_local` for process with name run_seqspec_modify_atac\n",
      "Dec-11 03:16:15.446 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Dec-11 03:16:15.446 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Dec-11 03:16:15.449 [main] DEBUG nextflow.executor.Executor - [warm up] executor > google-batch (fusion enabled)\n",
      "Dec-11 03:16:15.454 [main] DEBUG n.processor.TaskPollingMonitor - Creating task monitor for executor 'google-batch' > capacity: 1000; pollInterval: 10s; dumpInterval: 5m \n",
      "Dec-11 03:16:15.457 [main] DEBUG n.processor.TaskPollingMonitor - >>> barrier register (monitor: google-batch)\n",
      "Dec-11 03:16:15.463 [main] DEBUG nextflow.cloud.google.GoogleOpts - Google auth via application DEFAULT\n",
      "Dec-11 03:16:15.468 [main] DEBUG n.c.google.batch.GoogleBatchExecutor - [GOOGLE BATCH] Executor config=BatchConfig[googleOpts=GoogleOpts(projectId:hai-gcp-genomic, credsFile:null, location:us-central1, enableRequesterPaysBuckets:false, httpConnectTimeout:1m, httpReadTimeout:1m, credentials:UserCredentials{requestMetadata=null, temporaryAccess=null, clientId=764086051850-6qr4p6gpi6hn506pt8ejuq83di341hur.apps.googleusercontent.com, refreshToken=1//04-ZrKvCfXPa5CgYIARAAGAQSNwF-L9IrkFtcwyfUDAhTeAquw0t0utu5nMRfG-U9l5uflp_XJI0EMWlO9aPKmi0CxdbKiKRekQc, tokenServerUri=https://oauth2.googleapis.com/token, transportFactoryClassName=com.google.auth.oauth2.OAuth2Utils$DefaultHttpTransportFactory, quotaProjectId=hai-gcp-genomic})\n",
      "Dec-11 03:16:15.487 [main] DEBUG n.c.google.batch.client.BatchClient - [GOOGLE BATCH] Creating service client with config credentials\n",
      "Dec-11 03:16:16.253 [main] INFO  nextflow.script.BaseScript - after run_seqspec_modify_atac\n",
      "Dec-11 03:16:16.258 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:seqspec_local` matches labels `seqspec_local` for process with name run_seqspec_print\n",
      "Dec-11 03:16:16.260 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Dec-11 03:16:16.260 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Dec-11 03:16:16.264 [main] INFO  nextflow.script.BaseScript - after run_seqspec_print\n",
      "Dec-11 03:16:16.265 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:chromap_local` matches labels `chromap_local` for process with name run_chromap_test\n",
      "Dec-11 03:16:16.266 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Dec-11 03:16:16.266 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Dec-11 03:16:16.274 [main] INFO  nextflow.script.BaseScript - after run_chromap_test\n",
      "Dec-11 03:16:16.293 [main] INFO  nextflow.script.BaseScript - after genome_fasta_ch\n",
      "Dec-11 03:16:16.294 [main] INFO  nextflow.script.BaseScript - after genome_gtf_ch\n",
      "Dec-11 03:16:16.303 [main] INFO  nextflow.script.BaseScript - after genome_chromap_idx download\n",
      "Dec-11 03:16:16.303 [main] INFO  nextflow.script.BaseScript - chromap parameters\n",
      "Dec-11 03:16:16.304 [main] INFO  nextflow.script.BaseScript - false\n",
      "Dec-11 03:16:16.305 [main] INFO  nextflow.script.BaseScript - false\n",
      "Dec-11 03:16:16.305 [main] INFO  nextflow.script.BaseScript - false\n",
      "Dec-11 03:16:16.305 [main] INFO  nextflow.script.BaseScript - false\n",
      "Dec-11 03:16:16.306 [main] INFO  nextflow.script.BaseScript - false\n",
      "Dec-11 03:16:16.307 [main] INFO  nextflow.script.BaseScript - false\n",
      "Dec-11 03:16:16.308 [main] INFO  nextflow.script.BaseScript - 20\n",
      "Dec-11 03:16:16.308 [main] INFO  nextflow.script.BaseScript - 20\n",
      "Dec-11 03:16:16.309 [main] INFO  nextflow.script.BaseScript - 0.20\n",
      "Dec-11 03:16:16.309 [main] INFO  nextflow.script.BaseScript - r1:0:-1,bc:0:-1\n",
      "Dec-11 03:16:16.309 [main] INFO  nextflow.script.BaseScript - false\n",
      "Dec-11 03:16:16.309 [main] INFO  nextflow.script.BaseScript - false\n",
      "Dec-11 03:16:16.310 [main] INFO  nextflow.script.BaseScript - path\n",
      "Dec-11 03:16:16.315 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:chromap_map_idx` matches labels `chromap_map_idx` for process with name run_chromap_map_to_idx\n",
      "Dec-11 03:16:16.318 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Dec-11 03:16:16.319 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Dec-11 03:16:16.326 [main] INFO  nextflow.script.BaseScript - finished run_chromap_map_to_idx\n",
      "Dec-11 03:16:16.326 [main] INFO  nextflow.script.BaseScript - before call bgzip\n",
      "Dec-11 03:16:16.327 [main] INFO  nextflow.script.BaseScript - DataflowBroadcast around DataflowStream[?]\n",
      "Dec-11 03:16:16.331 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:bgzip` matches labels `bgzip` for process with name run_bgzip\n",
      "Dec-11 03:16:16.333 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Dec-11 03:16:16.333 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Dec-11 03:16:16.336 [main] INFO  nextflow.script.BaseScript - after call bgzip\n",
      "Dec-11 03:16:16.336 [main] INFO  nextflow.script.BaseScript - run_tabix.sh\n",
      "Dec-11 03:16:16.336 [main] INFO  nextflow.script.BaseScript - before call run_tabix_fragment_file\n",
      "Dec-11 03:16:16.340 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tabix` matches labels `tabix` for process with name run_tabix\n",
      "Dec-11 03:16:16.341 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Dec-11 03:16:16.341 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Dec-11 03:16:16.345 [main] INFO  nextflow.script.BaseScript - after call run_tabix_fragment_file\n",
      "Dec-11 03:16:16.345 [main] INFO  nextflow.script.BaseScript - before call run_merge_logs\n",
      "Dec-11 03:16:16.348 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:merge_log` matches labels `merge_log` for process with name run_merge_logs\n",
      "Dec-11 03:16:16.349 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Dec-11 03:16:16.350 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Dec-11 03:16:16.355 [main] INFO  nextflow.script.BaseScript - after call run_merge_logs\n",
      "Dec-11 03:16:16.355 [main] INFO  nextflow.script.BaseScript - before call run_merge_logs\n",
      "Dec-11 03:16:16.356 [main] INFO  nextflow.script.BaseScript - before call run_filter_fragments\n",
      "Dec-11 03:16:16.359 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:filter_fragments` matches labels `filter_fragments` for process with name run_filter_fragments\n",
      "Dec-11 03:16:16.362 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Dec-11 03:16:16.362 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Dec-11 03:16:16.369 [main] INFO  nextflow.script.BaseScript - after call run_filter_fragments\n",
      "Dec-11 03:16:16.369 [main] INFO  nextflow.script.BaseScript - before call run_tabix for the second time with filtered_fragment_file_out\n",
      "Dec-11 03:16:16.370 [main] INFO  nextflow.script.BaseScript - DataflowBroadcast around DataflowStream[?]\n",
      "Dec-11 03:16:16.373 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tabix_after_fragments` matches labels `tabix_after_fragments` for process with name run_tabix_filtered_fragments\n",
      "Dec-11 03:16:16.374 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Dec-11 03:16:16.374 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Dec-11 03:16:16.378 [main] INFO  nextflow.script.BaseScript - before call run_tss\n",
      "Dec-11 03:16:16.381 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tss` matches labels `tss` for process with name run_tss\n",
      "Dec-11 03:16:16.382 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Dec-11 03:16:16.382 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Dec-11 03:16:16.385 [main] INFO  nextflow.script.BaseScript - after call run_tss\n",
      "Dec-11 03:16:16.387 [main] DEBUG nextflow.Session - Workflow process names [dsl2]: run_merge_logs, run_seqspec_modify_rna, run_seqspec_print, run_tss, run_bgzip, run_tabix, run_filter_fragments, run_create_chromap_idx, run_seqspec_check, run_seqspec_test, run_seqspec_index_rna_kb, run_seqspec_modify_atac, run_chromap_test, run_tabix_filtered_fragments, run_downloadFiles, run_chromap_map_to_idx\n",
      "Dec-11 03:16:16.709 [main] INFO  io.seqera.tower.plugin.TowerClient - Monitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5R5XxgzcBSHmnW\n",
      "Dec-11 03:16:16.710 [main] DEBUG nextflow.Session - Igniting dataflow network (15)\n",
      "Dec-11 03:16:16.722 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_seqspec_modify_atac\n",
      "Dec-11 03:16:16.723 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_seqspec_print\n",
      "Dec-11 03:16:16.724 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_chromap_test\n",
      "Dec-11 03:16:16.726 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_chromap_map_to_idx\n",
      "Dec-11 03:16:16.727 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_bgzip\n",
      "Dec-11 03:16:16.727 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_tabix\n",
      "Dec-11 03:16:16.728 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_merge_logs\n",
      "Dec-11 03:16:16.729 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_filter_fragments\n",
      "Dec-11 03:16:16.729 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_tabix_filtered_fragments\n",
      "Dec-11 03:16:16.736 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_tss\n",
      "Dec-11 03:16:16.738 [main] DEBUG nextflow.script.ScriptRunner - Parsed script files:\n",
      "  Script_62ae1c49adcd151d: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/nf_wf_fg_singlecell_atac.nf\n",
      "  Script_f63f7a0583acd4b8: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_bgzip.nf\n",
      "  Script_d121259533d5bb0f: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tabix.nf\n",
      "  Script_7180cc8fa5a6505c: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_chromap_utils.nf\n",
      "  Script_e112646bcdb899f0: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_seqspec_utils.nf\n",
      "  Script_9d4639c8cd883c70: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_download_url_files.nf\n",
      "  Script_9c2ce54dcf0b5c06: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_merge_logs.nf\n",
      "  Script_e882890b13e4c102: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tss.nf\n",
      "  Script_88ab30021a80a861: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_filter_fragments.nf\n",
      "Dec-11 03:16:16.738 [main] DEBUG nextflow.script.ScriptRunner - > Awaiting termination \n",
      "Dec-11 03:16:16.738 [main] DEBUG nextflow.Session - Session await\n",
      "Dec-11 03:16:16.913 [Actor Thread 7] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Dec-11 03:16:16.915 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null, dryRunMode:false)\n",
      "Dec-11 03:16:16.942 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave assets bundle bin resources: /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin\n",
      "Dec-11 03:16:17.103 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave request container config: https://fusionfs.seqera.io/releases/v2.2-amd64.json\n",
      "Dec-11 03:16:17.333 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave container config response: [200] {\n",
      "  \"entrypoint\": [ \"/usr/bin/fusion\" ],\n",
      "  \"layers\": [\n",
      "    {\n",
      "      \"location\": \"https://fusionfs.seqera.io/releases/pkg/2/2/10/fusion-amd64.tar.gz\",\n",
      "      \"gzipDigest\": \"sha256:2e10d6a7e2d43592f25f20fbe42d3876e9c78412f8380cdb6ddc80420cae3f6c\",\n",
      "      \"gzipSize\": 30260552,\n",
      "      \"tarDigest\": \"sha256:27df049324b6b879b593d2f0afa4e676331b364ec134ca59348934af89c5e508\",\n",
      "      \"skipHashing\": true\n",
      "    }\n",
      "  ]\n",
      "}\n",
      "\n",
      "Dec-11 03:16:17.447 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/chromap_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+1XW3Ob...; tarDigest=sha256:024e62696f1ddef937ef21e9fb6fdd23db8c5463f001e76a76b323089d30f410; gzipDigest=sha256:a447c80a9d97403d63f9011aef26eccfdf65f1b932a232b8d9f274aab4167aa8; gzipSize=1059], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/10/fusion-amd64.tar.gz; tarDigest=sha256:27df049324b6b879b593d2f0afa4e676331b364ec134ca59348934af89c5e508; gzipDigest=sha256:2e10d6a7e2d43592f25f20fbe42d3876e9c78412f8380cdb6ddc80420cae3f6c; gzipSize=30260552]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2023-12-11T03:16:17.446221Z, fingerprint:29a62ae6c13335c76388d1f486b9b14d, freeze:false, format:null, dryRun:false)\n",
      "Dec-11 03:16:18.046 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"77c814961eb0\",\"targetImage\":\"wave.seqera.io/wt/77c814961eb0/eilalan/chromap_env:latest\",\"expiration\":\"2023-12-12T15:16:17.993794813Z\",\"containerImage\":\"docker.io/eilalan/chromap_env:latest\"}\n",
      "Dec-11 03:16:18.741 [Actor Thread 7] INFO  nextflow.processor.TaskProcessor - [db/6ecefc] Cached process > run_chromap_test\n",
      "Dec-11 03:16:20.634 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/fastqs/ATAC/atac_R1_SRR18677642.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-f7e37c4b-c1ab-4ccb-a8cb-153d209e8602/52/d8d73bad2654a6c22f23717483c056/atac_R1_SRR18677642.fastq.gz\n",
      "Dec-11 03:16:20.688 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/fastqs/ATAC/atac_R3_SRR18677642.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-f7e37c4b-c1ab-4ccb-a8cb-153d209e8602/2e/49ad06aa6cd6d7bd59d457ac6055f3/atac_R3_SRR18677642.fastq.gz\n",
      "Dec-11 03:16:20.688 [FileTransfer-2] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/fastqs/ATAC/atac_R2_SRR18677642.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-f7e37c4b-c1ab-4ccb-a8cb-153d209e8602/63/dcbbccff592e0ab8b16bc0526e922d/atac_R2_SRR18677642.fastq.gz\n",
      "Dec-11 03:16:20.728 [FileTransfer-4] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/nf_spec_dogmaseq_dig.yaml at gs://eila-nextflow-bucket/work/ATAC/stage-f7e37c4b-c1ab-4ccb-a8cb-153d209e8602/02/6f0e96773aa7bc71988792e5c9784c/nf_spec_dogmaseq_dig.yaml\n",
      "Dec-11 03:16:20.888 [FileTransfer-5] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/RNA-737K-arc-v1.txt at gs://eila-nextflow-bucket/work/ATAC/stage-f7e37c4b-c1ab-4ccb-a8cb-153d209e8602/ab/8fd9454acd38f552158cbf8dfc15ff/RNA-737K-arc-v1.txt\n",
      "Dec-11 03:16:20.897 [FileTransfer-6] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/specs/dogmaseq-dig/Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index at gs://eila-nextflow-bucket/work/ATAC/stage-f7e37c4b-c1ab-4ccb-a8cb-153d209e8602/8c/71d86079facc3c45344e2052f27c6d/Homo_sapiens.GRCh38.dna.primary_assembly.chromap.ref.index\n",
      "Dec-11 03:16:20.901 [Actor Thread 11] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/seqspec_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+1XW3Ob...; tarDigest=sha256:024e62696f1ddef937ef21e9fb6fdd23db8c5463f001e76a76b323089d30f410; gzipDigest=sha256:a447c80a9d97403d63f9011aef26eccfdf65f1b932a232b8d9f274aab4167aa8; gzipSize=1059], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/10/fusion-amd64.tar.gz; tarDigest=sha256:27df049324b6b879b593d2f0afa4e676331b364ec134ca59348934af89c5e508; gzipDigest=sha256:2e10d6a7e2d43592f25f20fbe42d3876e9c78412f8380cdb6ddc80420cae3f6c; gzipSize=30260552]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2023-12-11T03:16:20.900648Z, fingerprint:146b8bdc95b045765d335d8f55978d2b, freeze:false, format:null, dryRun:false)\n",
      "Dec-11 03:16:20.904 [FileTransfer-7] DEBUG nextflow.file.FilePorter - Local cache found for foreign file http://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz at gs://eila-nextflow-bucket/work/ATAC/stage-f7e37c4b-c1ab-4ccb-a8cb-153d209e8602/f3/0b877a714c8f38a60dbb9221529e08/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\n",
      "Dec-11 03:16:21.013 [Actor Thread 11] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"15552cae4d82\",\"targetImage\":\"wave.seqera.io/wt/15552cae4d82/eilalan/seqspec_env:latest\",\"expiration\":\"2023-12-12T15:16:20.960998889Z\",\"containerImage\":\"docker.io/eilalan/seqspec_env:latest\"}\n",
      "Dec-11 03:16:22.252 [Actor Thread 11] INFO  nextflow.processor.TaskProcessor - [c8/7e4656] Cached process > run_seqspec_modify_atac (1)\n",
      "Dec-11 03:16:22.303 [Actor Thread 9] INFO  nextflow.processor.TaskProcessor - [05/dd7a46] Cached process > run_chromap_map_to_idx (1)\n",
      "Dec-11 03:16:22.325 [Actor Thread 2] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/utils_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+1XW3Ob...; tarDigest=sha256:024e62696f1ddef937ef21e9fb6fdd23db8c5463f001e76a76b323089d30f410; gzipDigest=sha256:a447c80a9d97403d63f9011aef26eccfdf65f1b932a232b8d9f274aab4167aa8; gzipSize=1059], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/10/fusion-amd64.tar.gz; tarDigest=sha256:27df049324b6b879b593d2f0afa4e676331b364ec134ca59348934af89c5e508; gzipDigest=sha256:2e10d6a7e2d43592f25f20fbe42d3876e9c78412f8380cdb6ddc80420cae3f6c; gzipSize=30260552]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2023-12-11T03:16:22.324849Z, fingerprint:120f380ab97eac3213d6ae718a45b594, freeze:false, format:null, dryRun:false)\n",
      "Dec-11 03:16:22.545 [Actor Thread 2] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"d82788f7d3e7\",\"targetImage\":\"wave.seqera.io/wt/d82788f7d3e7/eilalan/utils_env:latest\",\"expiration\":\"2023-12-12T15:16:22.493519529Z\",\"containerImage\":\"docker.io/eilalan/utils_env:latest\"}\n",
      "Dec-11 03:16:22.982 [Actor Thread 7] INFO  nextflow.processor.TaskProcessor - [25/135075] Cached process > run_seqspec_print (1)\n",
      "Dec-11 03:16:23.204 [Actor Thread 2] INFO  nextflow.processor.TaskProcessor - [7b/9d8290] Cached process > run_bgzip (1)\n",
      "Dec-11 03:16:23.386 [Actor Thread 5] INFO  nextflow.processor.TaskProcessor - [1e/4afc9b] Cached process > run_merge_logs (1)\n",
      "Dec-11 03:16:23.994 [Actor Thread 12] INFO  nextflow.processor.TaskProcessor - [87/dcf012] Cached process > run_tabix (1)\n",
      "Dec-11 03:16:24.222 [FileTransfer-8] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/nf_wf_fg_singlecell/na.na at gs://eila-nextflow-bucket/work/ATAC/stage-f7e37c4b-c1ab-4ccb-a8cb-153d209e8602/fa/e248d3ca5c1f9fcb994134d160cc35/na.na\n",
      "Dec-11 03:16:25.048 [Actor Thread 8] INFO  nextflow.processor.TaskProcessor - [e1/008d5d] Cached process > run_filter_fragments (1)\n",
      "Dec-11 03:16:26.032 [Actor Thread 12] INFO  nextflow.processor.TaskProcessor - [c1/1f5aad] Cached process > run_tabix_filtered_fragments (1)\n",
      "Dec-11 03:16:26.045 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/tss_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+1XW3Ob...; tarDigest=sha256:024e62696f1ddef937ef21e9fb6fdd23db8c5463f001e76a76b323089d30f410; gzipDigest=sha256:a447c80a9d97403d63f9011aef26eccfdf65f1b932a232b8d9f274aab4167aa8; gzipSize=1059], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/10/fusion-amd64.tar.gz; tarDigest=sha256:27df049324b6b879b593d2f0afa4e676331b364ec134ca59348934af89c5e508; gzipDigest=sha256:2e10d6a7e2d43592f25f20fbe42d3876e9c78412f8380cdb6ddc80420cae3f6c; gzipSize=30260552]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2023-12-11T03:16:26.044711Z, fingerprint:5205411f641f099aa5dcb0695c20bcbf, freeze:false, format:null, dryRun:false)\n",
      "Dec-11 03:16:26.153 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"273a847a6d20\",\"targetImage\":\"wave.seqera.io/wt/273a847a6d20/eilalan/tss_env:latest\",\"expiration\":\"2023-12-12T15:16:26.101038142Z\",\"containerImage\":\"docker.io/eilalan/tss_env:latest\"}\n",
      "Dec-11 03:16:27.343 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_tss (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Dec-11 03:16:28.867 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_tss (1)` submitted > job=nf-0db92d4b-1702264586694; uid=nf-0db92d4b-170226-55398da7-042a-49d80; work-dir=gs://eila-nextflow-bucket/work/ATAC/0d/b92d4b523d1aab1a4b799fb3b8a7fd\n",
      "Dec-11 03:16:28.867 [Task submitter] INFO  nextflow.Session - [0d/b92d4b] Submitted process > run_tss (1)\n",
      "Dec-11 03:19:25.522 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_tss (1)` - terminated job=nf-0db92d4b-1702264586694; state=SUCCEEDED\n",
      "Dec-11 03:19:25.838 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 10; name: run_tss (1); status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/0d/b92d4b523d1aab1a4b799fb3b8a7fd]\n",
      "Dec-11 03:19:26.064 [Task monitor] DEBUG nextflow.processor.TaskProcessor - Process `run_tss (1)` is unable to find [CloudStoragePath]: `/work/ATAC/0d/b92d4b523d1aab1a4b799fb3b8a7fd/filtered_fragment_file.bed.tbi.tss` (pattern: `filtered_fragment_file.bed.tbi.tss`)\n",
      "Dec-11 03:19:26.071 [Task monitor] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for\n",
      "  task: name=run_tss (1); work-dir=gs://eila-nextflow-bucket/work/ATAC/0d/b92d4b523d1aab1a4b799fb3b8a7fd\n",
      "  error [nextflow.exception.MissingFileException]: Missing output file(s) `filtered_fragment_file.bed.tbi.tss` expected by process `run_tss (1)`\n",
      "Dec-11 03:19:26.434 [Task monitor] ERROR nextflow.processor.TaskProcessor - Error executing process > 'run_tss (1)'\n",
      "\n",
      "Caused by:\n",
      "  Missing output file(s) `filtered_fragment_file.bed.tbi.tss` expected by process `run_tss (1)`\n",
      "\n",
      "Command executed:\n",
      "\n",
      "  echo 'start run_tss'\n",
      "  ls /usr/local/bin/\n",
      "  \n",
      "  echo 'finished run_tss'\n",
      "\n",
      "Command exit status:\n",
      "  0\n",
      "\n",
      "Command output:\n",
      "  start run_tss\n",
      "  atac_barcode_and_subpool.sh\n",
      "  create_sample_filtered_fragment_file.sh\n",
      "  run_tabix.sh\n",
      "  finished run_tss\n",
      "\n",
      "Work dir:\n",
      "  gs://eila-nextflow-bucket/work/ATAC/0d/b92d4b523d1aab1a4b799fb3b8a7fd\n",
      "\n",
      "Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`\n",
      "Dec-11 03:19:26.438 [main] DEBUG nextflow.Session - Session await > all processes finished\n",
      "Dec-11 03:19:26.695 [Task monitor] DEBUG nextflow.Session - Session aborted -- Cause: Missing output file(s) `filtered_fragment_file.bed.tbi.tss` expected by process `run_tss (1)`\n",
      "Dec-11 03:19:26.997 [main] DEBUG nextflow.Session - Session await > all barriers passed\n",
      "Dec-11 03:19:27.000 [Task monitor] DEBUG n.processor.TaskPollingMonitor - <<< barrier arrives (monitor: google-batch) - terminating tasks monitor poll loop\n",
      "Dec-11 03:19:27.362 [main] DEBUG n.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=0; failedCount=1; ignoredCount=0; cachedCount=9; pendingCount=0; submittedCount=0; runningCount=0; retriesCount=0; abortedCount=0; succeedDuration=0ms; failedDuration=90ms; cachedDuration=2m 49s;loadCpus=0; loadMemory=0; peakRunning=1; peakCpus=1; peakMemory=0; ]\n",
      "Dec-11 03:19:27.793 [main] DEBUG nextflow.cache.CacheDB - Closing CacheDB done\n",
      "Dec-11 03:19:27.793 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-wave@1.0.0'\n",
      "Dec-11 03:19:27.793 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-wave\n",
      "Dec-11 03:19:27.793 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-tower@1.6.3'\n",
      "Dec-11 03:19:27.793 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-tower\n",
      "Dec-11 03:19:27.793 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-google@1.8.3'\n",
      "Dec-11 03:19:27.793 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-google\n",
      "Dec-11 03:19:27.818 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye\n"
     ]
    }
   ],
   "source": [
    "!cat .nextflow.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile -resume\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile -resume\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile google_batch_profile -resume\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile -resume\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow.config -profile docker_profile -resume\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow.config -profile google_batch_profile -resume"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !cat .nextflow.log "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !ls /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/seqspec/specs/dogmaseq-dig/fastqs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !nextflow run /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_processes/nf_prcs_print.nf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !ls /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_processes/nf_prcs_print.nf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !cat .command.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !nextflow run nf_wf_fg_singlecell.nf -c nextflow.config -with-docker eilalan/env_synapse:latest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
