{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "l2-LZ9798Ibb"
   },
   "source": [
    "# Building Count Matrices with `cellatlas`\n",
    "\n",
    "A major challenge in uniformly preprocessing large amounts of single-cell genomics data from a variety of different assays is identifying and handling sequenced elements in a coherent and consistent fashion. Cell barcodes in reads from RNAseq data from 10x Multiome, for example, must be extracted and error corrected in the manner as cell barcodes in reads from ATACseq data from 10xMultiome so that barcode-barcode registration can occur. Uniform processing in this way minimzes computational variability and enables cross-assay comparisons.\n",
    "\n",
    "In this notebook we demonstrate how single-cell genomics data can be preprocessed to generate a cell by feature count matrix. This requires:\n",
    "\n",
    "1. FASTQ files\n",
    "2. `seqspec` specification for the FASTQ files\n",
    "3. Genome Sequence FASTA\n",
    "4. Genome Annotation GTF\n",
    "5. (optional) Feature barcode list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "-QBK2zWg_AI1"
   },
   "source": [
    "-------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "iKTmVa2A_BNr"
   },
   "source": [
    "## Install packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "2VTXQLMD2Y6m",
    "outputId": "0b7451bc-416e-49ce-a50d-4e2906aaf09d"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tree-2.1.0.tgz      100%[===================>]  57.80K  --.-KB/s    in 0.1s    \n",
      "gcc -O3 -std=c11 -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -c -o tree.o tree.c\n",
      "gcc -O3 -std=c11 -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -c -o list.o list.c\n",
      "gcc -O3 -std=c11 -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -c -o hash.o hash.c\n",
      "gcc -O3 -std=c11 -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -c -o color.o color.c\n",
      "gcc -O3 -std=c11 -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -c -o file.o file.c\n",
      "gcc -O3 -std=c11 -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -c -o filter.o filter.c\n",
      "gcc -O3 -std=c11 -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -c -o info.o info.c\n",
      "gcc -O3 -std=c11 -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -c -o unix.o unix.c\n",
      "gcc -O3 -std=c11 -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -c -o xml.o xml.c\n",
      "gcc -O3 -std=c11 -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -c -o json.o json.c\n",
      "gcc -O3 -std=c11 -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -c -o html.o html.c\n",
      "gcc -O3 -std=c11 -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -c -o strverscmp.o strverscmp.c\n",
      "gcc  -o tree tree.o list.o hash.o color.o file.o filter.o info.o unix.o xml.o json.o html.o strverscmp.o\n",
      "jq-linux64          100%[===================>]   3.77M  --.-KB/s    in 0.06s   \n",
      "Cloning into 'cellatlas'...\n",
      "remote: Enumerating objects: 663, done.\u001b[K\n",
      "remote: Counting objects: 100% (151/151), done.\u001b[K\n",
      "remote: Compressing objects: 100% (92/92), done.\u001b[K\n",
      "remote: Total 663 (delta 95), reused 104 (delta 59), pack-reused 512\u001b[K\n",
      "Receiving objects: 100% (663/663), 804.38 MiB | 34.87 MiB/s, done.\n",
      "Resolving deltas: 100% (396/396), done.\n",
      "Updating files: 100% (122/122), done.\n",
      "Proceed (Y/n)? "
     ]
    }
   ],
   "source": [
    "# Install `tree` to view files\n",
    "!wget --quiet --show-progress ftp://mama.indstate.edu/linux/tree/tree-2.1.0.tgz\n",
    "!tar -xf tree-2.1.0.tgz && cd tree-2.1.0 && make -j16 && make install > /dev/null\n",
    "\n",
    "# Install `jq, a command-line tool for extracting key value pairs from JSON files\n",
    "!wget --quiet --show-progress https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64\n",
    "!chmod +x jq-linux64 && mv jq-linux64 /usr/local/bin/jq\n",
    "\n",
    "# Clone the cellatlas repo and install the package\n",
    "!git clone https://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@github.com/cellatlas/cellatlas.git > /dev/null\n",
    "!cd cellatlas && pip install --quiet . > /dev/null\n",
    "\n",
    "# Install dependencies\n",
    "!yes | pip uninstall --quiet seqspec\n",
    "!pip install --quiet git+https://github.com/IGVF/seqspec.git > /dev/null\n",
    "!pip install --quiet gget kb-python > /dev/null"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "6RhzGyP9_cEu"
   },
   "source": [
    "**Note:** We move the relevant data to the working directory and `gunzip` the barcode onlist."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "id": "vbIt2r2A3hbP"
   },
   "outputs": [],
   "source": [
    "!mv cellatlas/examples/multi-dogmaseq-lll/* .\n",
    "!gunzip *.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "1ievY-64-u43"
   },
   "source": [
    "## Preprocessing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "OrUgMsqp-weB"
   },
   "source": [
    "### Example the spec\n",
    "\n",
    "We first use `seqspec print` to check that the read structure matches what we expect. This command prints out an ordered tree representation of the sequenced elements contained in the FASTQ files. Note that the names of the nodes in the `seqspec` must match the names of the FASTQ files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "oxK70t3e718z",
    "outputId": "2ef4c73e-2907-4a79-eb0f-364c605d9019"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                                                                      ┌─'protein_cell_bc:16'\n",
      "                                                                    ┌─protein_R1_SRR18677632.fastq.gz─┤\n",
      "                                  ┌─protein─────────────────────────┤                                 └─'protein_umi:12'\n",
      "                                  │                                 └─protein_R2_SRR18677632.fastq.gz ──'protein:15'\n",
      "                                  │                                                                   ┌─'rna_cell_bc:16'\n",
      "                                  │                                 ┌─rna_R1_SRR18677629.fastq.gz─────┤\n",
      "──────────────────────────────────┼─rna─────────────────────────────┤                                 └─'rna_umi:12'\n",
      "                                  │                                 └─rna_R2_SRR18677629.fastq.gz──── ──'cdna:102'\n",
      "                                  │                                 ┌─atac_R1_SRR18677633.fastq.gz─── ──'gDNA-1:52'\n",
      "                                  └─atac────────────────────────────┼─atac_R2_SRR18677633.fastq.gz─── ──'atac_cell_bc:16'\n",
      "                                                                    └─atac_R3_SRR18677633.fastq.gz─── ──'gDNA-2:52'\n"
     ]
    }
   ],
   "source": [
    "!seqspec print spec.yaml"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "OExrOQ4f-xoy"
   },
   "source": [
    "### Fetch the references\n",
    "This step is only necessary if modality that we are processing uses a transcriptome reference-based alignment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "I63OT0t92weV",
    "outputId": "ebfccb3f-fa33-43ec-e0cb-b3b74e4b18b5"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sun Sep 24 20:54:14 2023 INFO Fetching reference information for homo_sapiens from Ensembl release: 110.\n"
     ]
    }
   ],
   "source": [
    "!gget ref -o ref.json -w dna,gtf homo_sapiens"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "id": "xpHlsfEA8x1m"
   },
   "outputs": [],
   "source": [
    "FA=!echo $(jq -r '.homo_sapiens.genome_dna.ftp' ref.json)\n",
    "\n",
    "GTF=!echo $(jq -r '.homo_sapiens.annotation_gtf.ftp' ref.json)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "NRmrjlw9-Ls0",
    "outputId": "0e31b301-ad7e-4e1b-9285-215542f6c85b"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['http://ftp.ensembl.org/pub/release-110/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz']"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "Wf2-MKlR-U-c",
    "outputId": "9dda216b-e203-4eb6-a319-44091c2e5765"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['http://ftp.ensembl.org/pub/release-110/gtf/homo_sapiens/Homo_sapiens.GRCh38.110.gtf.gz']"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GTF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 35
    },
    "id": "9bbw7Ph_-Q5n",
    "outputId": "e8d94840-2dd9-4e24-9817-347c48e60535"
   },
   "outputs": [
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "type": "string"
      },
      "text/plain": [
       "'http://ftp.ensembl.org/pub/release-110/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz'"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FA=FA[0]\n",
    "FA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 35
    },
    "id": "lRDEUYJd-T2d",
    "outputId": "5569f774-bb50-4887-d746-43d89fdf0a0d"
   },
   "outputs": [
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "type": "string"
      },
      "text/plain": [
       "'http://ftp.ensembl.org/pub/release-110/gtf/homo_sapiens/Homo_sapiens.GRCh38.110.gtf.gz'"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GTF=GTF[0]\n",
    "GTF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "hAa6kDU5-zak"
   },
   "source": [
    "### Build the pipeline\n",
    "\n",
    "We now supply all of the relevant objects to `cellatlas build` to produce the appropriate commands to be run to build the pipeline. This includes a reference building step and a read counting and quantification step both of which are performed with `kallisto` and `bustools` as part of the `kb-python` package."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "id": "53NxlmVO2Ob_"
   },
   "outputs": [],
   "source": [
    "# protein\n",
    "!cellatlas build -o protein_cellatlas_out \\\n",
    "-m protein \\\n",
    "-s spec.yaml \\\n",
    "-fa $FA -g $GTF \\\n",
    "-fb protein_feature_barcodes.txt \\\n",
    "fastqs/protein_R1_SRR18677632.fastq.gz fastqs/protein_R2_SRR18677632.fastq.gz\n",
    "\n",
    "# rna\n",
    "!cellatlas build -o rna_cellatlas_out \\\n",
    "-m rna \\\n",
    "-s spec.yaml \\\n",
    "-fa $FA -g $GTF \\\n",
    "-fb protein_feature_barcodes.txt \\\n",
    "fastqs/rna_R1_SRR18677629.fastq.gz fastqs/rna_R2_SRR18677629.fastq.gz\n",
    "\n",
    "# atac\n",
    "!cellatlas build -o atac_cellatlas_out \\\n",
    "-m atac \\\n",
    "-s spec.yaml \\\n",
    "-fa $FA -g $GTF \\\n",
    "-fb protein_feature_barcodes.txt \\\n",
    "fastqs/atac_R1_SRR18677633.fastq.gz fastqs/atac_R2_SRR18677633.fastq.gz fastqs/atac_R3_SRR18677633.fastq.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "2CvtZOUDAjfN"
   },
   "source": [
    "**Note**: The commands generated by `cellatlas build` are stored in the `out/cellatlas_info.json` file. We can view this file with `jq`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "049xtISb3sh4",
    "outputId": "8b51ef28-ec43-4faf-d5fb-3765e4534367"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### PROTEIN\n",
      "kb ref --workflow kite -i protein_cellatlas_out/index.idx -g protein_cellatlas_out/t2g.txt -f1 protein_cellatlas_out/transcriptome.fa protein_feature_barcodes.txt\n",
      "kb count --workflow kite -i protein_cellatlas_out/index.idx -g protein_cellatlas_out/t2g.txt -x 0,0,16:0,16,28:1,0,15 -w RNA-737K-arc-v1.txt -o protein_cellatlas_out --h5ad -t 2 fastqs/protein_R1_SRR18677632.fastq.gz fastqs/protein_R2_SRR18677632.fastq.gz\n",
      "\n",
      "\n",
      "#### RNA\n",
      "kb ref -i rna_cellatlas_out/index.idx -g rna_cellatlas_out/t2g.txt -f1 rna_cellatlas_out/transcriptome.fa http://ftp.ensembl.org/pub/release-110/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz http://ftp.ensembl.org/pub/release-110/gtf/homo_sapiens/Homo_sapiens.GRCh38.110.gtf.gz\n",
      "kb count -i rna_cellatlas_out/index.idx -g rna_cellatlas_out/t2g.txt -x 0,0,16:0,16,28:1,0,102 -w RNA-737K-arc-v1.txt -o rna_cellatlas_out --h5ad -t 2 fastqs/rna_R1_SRR18677629.fastq.gz fastqs/rna_R2_SRR18677629.fastq.gz\n",
      "\n",
      "\n",
      "#### ATAC\n",
      "minimap2 -d atac_cellatlas_out/ref.mmi http://ftp.ensembl.org/pub/release-110/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\n",
      "minimap2 -o atac_cellatlas_out/genome.0.sam -a -x sr -t 32 atac_cellatlas_out/ref.mmi \n",
      "samtools view -@ 8 -o atac_cellatlas_out/genome.u.0.bam -b atac_cellatlas_out/genome.0.sam\n",
      "samtools sort -@ 8 -o atac_cellatlas_out/genome.0.bam -n -m 8G atac_cellatlas_out/genome.u.0.bam\n",
      "Genrich -t atac_cellatlas_out/genome.0.bam -o atac_cellatlas_out/genome.0.bed -f atac_cellatlas_out/genome_peaks.0.log -v\n",
      "cat atac_cellatlas_out/genome.0.bed | bedtools sort | bedtools merge > atac_cellatlas_out/peaks.0.bed\n",
      "cat atac_cellatlas_out/peaks.*.bed | bedtools sort | bedtools merge > atac_cellatlas_out/peaks.bed\n",
      "zcat http://ftp.ensembl.org/pub/release-110/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz | fold -w 80 > atac_cellatlas_out/genome.fa\n",
      "bedtools getfasta -fi atac_cellatlas_out/genome.fa -bed atac_cellatlas_out/peaks.bed -fo atac_cellatlas_out/peaks.fa\n",
      "cat atac_cellatlas_out/peaks.fa | awk '{if($1~/>/)print $1\"\t\"$1\"\t\"$1}' > atac_cellatlas_out/t2g.txt\n",
      "sed -i 's/>//g' atac_cellatlas_out/t2g.txt\n",
      "kallisto index -i atac_cellatlas_out/index.idx atac_cellatlas_out/peaks.fa\n",
      "kb count -i atac_cellatlas_out/index.idx -g atac_cellatlas_out/t2g.txt -x 1,0,16:-1,-1,-1:0,0,52,2,0,52 -w ATA-737K-arc-v1.txt -o atac_cellatlas_out --h5ad -t 2 fastqs/atac_R1_SRR18677633.fastq.gz fastqs/atac_R2_SRR18677633.fastq.gz fastqs/atac_R3_SRR18677633.fastq.gz\n",
      "mkdir -p atac_cellatlas_out/counts_mult\n",
      "bustools count -o atac_cellatlas_out/counts_mult/cells_x_genes -g atac_cellatlas_out/t2g.txt -e atac_cellatlas_out/matrix.ec -t atac_cellatlas_out/transcripts.txt --genecounts --cm atac_cellatlas_out/output.unfiltered.bus\n"
     ]
    }
   ],
   "source": [
    "!echo \"#### PROTEIN\"\n",
    "!jq  -r '.commands[] | values[] | join(\"\\n\")' protein_cellatlas_out/cellatlas_info.json\n",
    "!printf \"\\n\\n\"\n",
    "\n",
    "!echo \"#### RNA\"\n",
    "!jq  -r '.commands[] | values[] | join(\"\\n\")' rna_cellatlas_out/cellatlas_info.json\n",
    "!printf \"\\n\\n\"\n",
    "\n",
    "!echo \"#### ATAC\"\n",
    "!jq  -r '.commands[] | values[] | join(\"\\n\")' atac_cellatlas_out/cellatlas_info.json"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "FNidpYNN-10f"
   },
   "source": [
    "### Run the pipeline\n",
    "\n",
    "To run the pipeline we simply extract the commands from `out/cellatlas_info.json` and pass them to `bash`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "7PDLF2Ps6wfz",
    "outputId": "ccc8e24f-c9fd-4058-e2d8-0b51489c800a"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2023-06-12 04:55:47,720]    INFO [ref_kite] Generating mismatch FASTA at protein_cellatlas_out/transcriptome.fa\n",
      "[2023-06-12 04:55:47,850]    INFO [ref_kite] Creating transcript-to-gene mapping at protein_cellatlas_out/t2g.txt\n",
      "[2023-06-12 04:55:47,888]    INFO [ref_kite] Indexing protein_cellatlas_out/transcriptome.fa to protein_cellatlas_out/index.idx\n",
      "[2023-06-12 04:55:59,893]    INFO [count] Using index protein_cellatlas_out/index.idx to generate BUS file to protein_cellatlas_out from\n",
      "[2023-06-12 04:55:59,893]    INFO [count]         fastqs/protein_R1_SRR18677632.fastq.gz\n",
      "[2023-06-12 04:55:59,893]    INFO [count]         fastqs/protein_R2_SRR18677632.fastq.gz\n",
      "[2023-06-12 04:56:03,215]    INFO [count] Sorting BUS file protein_cellatlas_out/output.bus to protein_cellatlas_out/tmp/output.s.bus\n",
      "[2023-06-12 04:56:13,536]    INFO [count] Inspecting BUS file protein_cellatlas_out/tmp/output.s.bus\n",
      "[2023-06-12 04:56:15,242]    INFO [count] Correcting BUS records in protein_cellatlas_out/tmp/output.s.bus to protein_cellatlas_out/tmp/output.s.c.bus with whitelist RNA-737K-arc-v1.txt\n",
      "[2023-06-12 04:56:18,455]    INFO [count] Sorting BUS file protein_cellatlas_out/tmp/output.s.c.bus to protein_cellatlas_out/output.unfiltered.bus\n",
      "[2023-06-12 04:56:24,276]    INFO [count] Generating count matrix protein_cellatlas_out/counts_unfiltered/cells_x_features from BUS file protein_cellatlas_out/output.unfiltered.bus\n",
      "[2023-06-12 04:56:25,386]    INFO [count] Reading matrix protein_cellatlas_out/counts_unfiltered/cells_x_features.mtx\n",
      "[2023-06-12 04:56:25,428]    INFO [count] Writing matrix to h5ad protein_cellatlas_out/counts_unfiltered/adata.h5ad\n"
     ]
    }
   ],
   "source": [
    "!jq  -r '.commands[] | values[] | join(\"\\n\")' protein_cellatlas_out/cellatlas_info.json | bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "7K_czSFQ4bPU",
    "outputId": "84c867c4-f33a-410f-b19d-a5f8657b7643"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2023-06-12 04:56:33,297]    INFO [ref] Preparing http://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz, http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz\n",
      "[2023-06-12 04:58:25,497]    INFO [ref] Splitting genome http://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz into cDNA at /content/tmp/tmp372ds2m1\n",
      "[2023-06-12 05:19:14,120]    INFO [ref] Concatenating 1 cDNAs to rna_cellatlas_out/transcriptome.fa\n",
      "[2023-06-12 05:19:16,196]    INFO [ref] Creating transcript-to-gene mapping at rna_cellatlas_out/t2g.txt\n",
      "[2023-06-12 05:19:19,351]    INFO [ref] Indexing rna_cellatlas_out/transcriptome.fa to rna_cellatlas_out/index.idx\n",
      "[2023-06-12 05:34:40,569]    INFO [count] Sorting BUS file rna_cellatlas_out/output.bus to rna_cellatlas_out/tmp/output.s.bus\n",
      "[2023-06-12 05:34:46,297]    INFO [count] Inspecting BUS file rna_cellatlas_out/tmp/output.s.bus\n",
      "[2023-06-12 05:34:48,302]    INFO [count] Correcting BUS records in rna_cellatlas_out/tmp/output.s.bus to rna_cellatlas_out/tmp/output.s.c.bus with whitelist RNA-737K-arc-v1.txt\n",
      "[2023-06-12 05:34:51,609]    INFO [count] Sorting BUS file rna_cellatlas_out/tmp/output.s.c.bus to rna_cellatlas_out/output.unfiltered.bus\n",
      "[2023-06-12 05:34:57,022]    INFO [count] Generating count matrix rna_cellatlas_out/counts_unfiltered/cells_x_genes from BUS file rna_cellatlas_out/output.unfiltered.bus\n",
      "[2023-06-12 05:35:03,972]    INFO [count] Reading matrix rna_cellatlas_out/counts_unfiltered/cells_x_genes.mtx\n",
      "[2023-06-12 05:35:05,425] WARNING [count] 20930 gene IDs do not have corresponding gene names. These genes will use their gene IDs instead.\n",
      "[2023-06-12 05:35:05,453]    INFO [count] Writing matrix to h5ad rna_cellatlas_out/counts_unfiltered/adata.h5ad\n"
     ]
    }
   ],
   "source": [
    "!jq  -r '.commands[] | values[] | join(\"\\n\")' rna_cellatlas_out/cellatlas_info.json | bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "blpvr8-44fAm"
   },
   "outputs": [],
   "source": [
    "!jq  -r '.commands[] | values[] | join(\"\\n\")' atac_cellatlas_out/cellatlas_info.json | bash"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "z5PcMU52A1DE"
   },
   "source": [
    "### Inspect outputs\n",
    "\n",
    "We inspect the `out/run_info.json` and `out/kb_info.json` as a simple QC on the pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "zCq2zEJD7cYf",
    "outputId": "5a352d56-0c60-4226-d592-8f4d9e04fe76"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      "\t\"n_targets\": 0,\n",
      "\t\"n_bootstraps\": 0,\n",
      "\t\"n_processed\": 1000000,\n",
      "\t\"n_pseudoaligned\": 1134,\n",
      "\t\"n_unique\": 1134,\n",
      "\t\"p_pseudoaligned\": 0.1,\n",
      "\t\"p_unique\": 0.1,\n",
      "\t\"kallisto_version\": \"0.48.0\",\n",
      "\t\"index_version\": 0,\n",
      "\t\"start_time\": \"Mon Jun 12 04:55:59 2023\",\n",
      "\t\"call\": \"/usr/local/lib/python3.10/dist-packages/kb_python/bins/linux/kallisto/kallisto bus -i protein_cellatlas_out/index.idx -o protein_cellatlas_out -x 0,0,16:0,16,28:1,0,15 -t 2 fastqs/protein_R1_SRR18677632.fastq.gz fastqs/protein_R2_SRR18677632.fastq.gz\"\n",
      "}\n",
      "{\n",
      "\t\"n_targets\": 0,\n",
      "\t\"n_bootstraps\": 0,\n",
      "\t\"n_processed\": 1000000,\n",
      "\t\"n_pseudoaligned\": 537164,\n",
      "\t\"n_unique\": 154572,\n",
      "\t\"p_pseudoaligned\": 53.7,\n",
      "\t\"p_unique\": 15.5,\n",
      "\t\"kallisto_version\": \"0.48.0\",\n",
      "\t\"index_version\": 0,\n",
      "\t\"start_time\": \"Mon Jun 12 05:33:41 2023\",\n",
      "\t\"call\": \"/usr/local/lib/python3.10/dist-packages/kb_python/bins/linux/kallisto/kallisto bus -i rna_cellatlas_out/index.idx -o rna_cellatlas_out -x 0,0,16:0,16,28:1,0,102 -t 2 fastqs/rna_R1_SRR18677629.fastq.gz fastqs/rna_R2_SRR18677629.fastq.gz\"\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "!cat protein_cellatlas_out/run_info.json\n",
    "!cat rna_cellatlas_out/run_info.json\n",
    "!cat atac_cellatlas_out/run_info.json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "AS8SyOhmH7LB",
    "outputId": "d76e5e1e-a71f-4bdf-d35b-48e27993e5df"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      "    \"workdir\": \"/content\",\n",
      "    \"version\": \"0.27.3\",\n",
      "    \"kallisto\": {\n",
      "        \"path\": \"/usr/local/lib/python3.10/dist-packages/kb_python/bins/linux/kallisto/kallisto\",\n",
      "        \"version\": \"0.48.0\"\n",
      "    },\n",
      "    \"bustools\": {\n",
      "        \"path\": \"/usr/local/lib/python3.10/dist-packages/kb_python/bins/linux/bustools/bustools\",\n",
      "        \"version\": \"0.41.0\"\n",
      "    },\n",
      "    \"start_time\": \"2023-06-12T04:55:57.788003\",\n",
      "    \"end_time\": \"2023-06-12T04:56:25.453003\",\n",
      "    \"elapsed\": 27.665,\n",
      "    \"call\": \"/usr/local/bin/kb count --workflow kite -i protein_cellatlas_out/index.idx -g protein_cellatlas_out/t2g.txt -x 0,0,16:0,16,28:1,0,15 -w RNA-737K-arc-v1.txt -o protein_cellatlas_out --h5ad -t 2 fastqs/protein_R1_SRR18677632.fastq.gz fastqs/protein_R2_SRR18677632.fastq.gz\",\n",
      "    \"commands\": [\n",
      "        \"kallisto bus -i protein_cellatlas_out/index.idx -o protein_cellatlas_out -x 0,0,16:0,16,28:1,0,15 -t 2 fastqs/protein_R1_SRR18677632.fastq.gz fastqs/protein_R2_SRR18677632.fastq.gz\",\n",
      "        \"bustools inspect protein_cellatlas_out/output.bus\",\n",
      "        \"bustools sort -o protein_cellatlas_out/tmp/output.s.bus -T protein_cellatlas_out/tmp -t 2 -m 4G protein_cellatlas_out/output.bus\",\n",
      "        \"bustools inspect protein_cellatlas_out/tmp/output.s.bus\",\n",
      "        \"bustools inspect -o protein_cellatlas_out/inspect.json -w RNA-737K-arc-v1.txt protein_cellatlas_out/tmp/output.s.bus\",\n",
      "        \"bustools correct -o protein_cellatlas_out/tmp/output.s.c.bus -w RNA-737K-arc-v1.txt protein_cellatlas_out/tmp/output.s.bus\",\n",
      "        \"bustools inspect protein_cellatlas_out/tmp/output.s.c.bus\",\n",
      "        \"bustools sort -o protein_cellatlas_out/output.unfiltered.bus -T protein_cellatlas_out/tmp -t 2 -m 4G protein_cellatlas_out/tmp/output.s.c.bus\",\n",
      "        \"bustools inspect protein_cellatlas_out/output.unfiltered.bus\",\n",
      "        \"bustools count -o protein_cellatlas_out/counts_unfiltered/cells_x_features -g protein_cellatlas_out/t2g.txt -e protein_cellatlas_out/matrix.ec -t protein_cellatlas_out/transcripts.txt --genecounts protein_cellatlas_out/output.unfiltered.bus\"\n",
      "    ],\n",
      "    \"runtimes\": [\n",
      "        2.3090341091156006,\n",
      "        1.0128636360168457,\n",
      "        9.217808961868286,\n",
      "        1.1022911071777344,\n",
      "        1.7053868770599365,\n",
      "        2.110013961791992,\n",
      "        1.1023950576782227,\n",
      "        4.717606544494629,\n",
      "        1.1023051738739014,\n",
      "        1.1022017002105713\n",
      "    ]\n",
      "}{\n",
      "    \"workdir\": \"/content\",\n",
      "    \"version\": \"0.27.3\",\n",
      "    \"kallisto\": {\n",
      "        \"path\": \"/usr/local/lib/python3.10/dist-packages/kb_python/bins/linux/kallisto/kallisto\",\n",
      "        \"version\": \"0.48.0\"\n",
      "    },\n",
      "    \"bustools\": {\n",
      "        \"path\": \"/usr/local/lib/python3.10/dist-packages/kb_python/bins/linux/bustools/bustools\",\n",
      "        \"version\": \"0.41.0\"\n",
      "    },\n",
      "    \"start_time\": \"2023-06-12T05:33:39.045108\",\n",
      "    \"end_time\": \"2023-06-12T05:35:05.614122\",\n",
      "    \"elapsed\": 86.569014,\n",
      "    \"call\": \"/usr/local/bin/kb count -i rna_cellatlas_out/index.idx -g rna_cellatlas_out/t2g.txt -x 0,0,16:0,16,28:1,0,102 -w RNA-737K-arc-v1.txt -o rna_cellatlas_out --h5ad -t 2 fastqs/rna_R1_SRR18677629.fastq.gz fastqs/rna_R2_SRR18677629.fastq.gz\",\n",
      "    \"commands\": [\n",
      "        \"kallisto bus -i rna_cellatlas_out/index.idx -o rna_cellatlas_out -x 0,0,16:0,16,28:1,0,102 -t 2 fastqs/rna_R1_SRR18677629.fastq.gz fastqs/rna_R2_SRR18677629.fastq.gz\",\n",
      "        \"bustools inspect rna_cellatlas_out/output.bus\",\n",
      "        \"bustools sort -o rna_cellatlas_out/tmp/output.s.bus -T rna_cellatlas_out/tmp -t 2 -m 4G rna_cellatlas_out/output.bus\",\n",
      "        \"bustools inspect rna_cellatlas_out/tmp/output.s.bus\",\n",
      "        \"bustools inspect -o rna_cellatlas_out/inspect.json -w RNA-737K-arc-v1.txt rna_cellatlas_out/tmp/output.s.bus\",\n",
      "        \"bustools correct -o rna_cellatlas_out/tmp/output.s.c.bus -w RNA-737K-arc-v1.txt rna_cellatlas_out/tmp/output.s.bus\",\n",
      "        \"bustools inspect rna_cellatlas_out/tmp/output.s.c.bus\",\n",
      "        \"bustools sort -o rna_cellatlas_out/output.unfiltered.bus -T rna_cellatlas_out/tmp -t 2 -m 4G rna_cellatlas_out/tmp/output.s.c.bus\",\n",
      "        \"bustools inspect rna_cellatlas_out/output.unfiltered.bus\",\n",
      "        \"bustools count -o rna_cellatlas_out/counts_unfiltered/cells_x_genes -g rna_cellatlas_out/t2g.txt -e rna_cellatlas_out/matrix.ec -t rna_cellatlas_out/transcripts.txt --genecounts rna_cellatlas_out/output.unfiltered.bus\"\n",
      "    ],\n",
      "    \"runtimes\": [\n",
      "        58.00415372848511,\n",
      "        1.4037292003631592,\n",
      "        4.308513402938843,\n",
      "        1.418806552886963,\n",
      "        2.004092216491699,\n",
      "        2.004195213317871,\n",
      "        1.3026580810546875,\n",
      "        4.109169006347656,\n",
      "        1.3022541999816895,\n",
      "        6.420598983764648\n",
      "    ]\n",
      "}"
     ]
    }
   ],
   "source": [
    "!cat protein_cellatlas_out/kb_info.json\n",
    "!cat rna_cellatlas_out/kb_info.json\n",
    "!cat atac_cellatlas_out/kb_info.json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "H5mVaYkTIT4x",
    "outputId": "212e458f-f5cf-41f8-cc3b-2cdbb74b7c5e"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[01;34mprotein_cellatlas_out\u001b[0m\n",
      "├── \u001b[00mcellatlas_info.json\u001b[0m\n",
      "├── \u001b[01;34mcounts_unfiltered\u001b[0m\n",
      "│   ├── \u001b[00madata.h5ad\u001b[0m\n",
      "│   ├── \u001b[00mcells_x_features.barcodes.txt\u001b[0m\n",
      "│   ├── \u001b[00mcells_x_features.genes.txt\u001b[0m\n",
      "│   └── \u001b[00mcells_x_features.mtx\u001b[0m\n",
      "├── \u001b[00mindex.idx\u001b[0m\n",
      "├── \u001b[00minspect.json\u001b[0m\n",
      "├── \u001b[00mkb_info.json\u001b[0m\n",
      "├── \u001b[00mmatrix.ec\u001b[0m\n",
      "├── \u001b[00moutput.bus\u001b[0m\n",
      "├── \u001b[00moutput.unfiltered.bus\u001b[0m\n",
      "├── \u001b[00mrun_info.json\u001b[0m\n",
      "├── \u001b[00mt2g.txt\u001b[0m\n",
      "├── \u001b[00mtranscriptome.fa\u001b[0m\n",
      "└── \u001b[00mtranscripts.txt\u001b[0m\n",
      "\n",
      "2 directories, 15 files\n",
      "\u001b[01;34mrna_cellatlas_out\u001b[0m\n",
      "├── \u001b[00mcellatlas_info.json\u001b[0m\n",
      "├── \u001b[01;34mcounts_unfiltered\u001b[0m\n",
      "│   ├── \u001b[00madata.h5ad\u001b[0m\n",
      "│   ├── \u001b[00mcells_x_genes.barcodes.txt\u001b[0m\n",
      "│   ├── \u001b[00mcells_x_genes.genes.txt\u001b[0m\n",
      "│   └── \u001b[00mcells_x_genes.mtx\u001b[0m\n",
      "├── \u001b[00mindex.idx\u001b[0m\n",
      "├── \u001b[00minspect.json\u001b[0m\n",
      "├── \u001b[00mkb_info.json\u001b[0m\n",
      "├── \u001b[00mmatrix.ec\u001b[0m\n",
      "├── \u001b[00moutput.bus\u001b[0m\n",
      "├── \u001b[00moutput.unfiltered.bus\u001b[0m\n",
      "├── \u001b[00mrun_info.json\u001b[0m\n",
      "├── \u001b[00mt2g.txt\u001b[0m\n",
      "├── \u001b[00mtranscriptome.fa\u001b[0m\n",
      "└── \u001b[00mtranscripts.txt\u001b[0m\n",
      "\n",
      "2 directories, 15 files\n"
     ]
    }
   ],
   "source": [
    "!tree protein_cellatlas_out\n",
    "!tree rna_cellatlas_out\n",
    "# !tree atac_cellatlas_out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "-udjiJdTRWu6"
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "colab": {
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
